<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>API Documentation &#8212; PyWGCNA  documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="_static/bootstrap-sphinx.css?v=4468db6d" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css?v=eafc0fe6" />
    <script src="_static/documentation_options.js?v=5929fcd5"></script>
    <script src="_static/doctools.js?v=888ff710"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Citation" href="cite.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">
<script type="text/javascript" src="_static/js/jquery-1.12.4.min.js"></script>
<script type="text/javascript" src="_static/js/jquery-fix.js"></script>
<script type="text/javascript" src="_static/bootstrap-3.4.1/js/bootstrap.min.js"></script>
<script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>

  </head><body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html">
          PyWGCNA</a>
        <span class="navbar-text navbar-version pull-left"><b></b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="suggested_reading.html">Suggested Reading</a></li>
<li class="toctree-l1"><a class="reference internal" href="cite.html">Citation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">API Documentation</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">API Documentation</a><ul>
<li><a class="reference internal" href="#PyWGCNA.geneExp.GeneExp"><code class="docutils literal notranslate"><span class="pre">GeneExp</span></code></a><ul>
<li><a class="reference internal" href="#PyWGCNA.geneExp.GeneExp.updateGeneInfo"><code class="docutils literal notranslate"><span class="pre">GeneExp.updateGeneInfo()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.geneExp.GeneExp.updateSampleInfo"><code class="docutils literal notranslate"><span class="pre">GeneExp.updateSampleInfo()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA"><code class="docutils literal notranslate"><span class="pre">WGCNA</span></code></a><ul>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.CalculateSignedKME"><code class="docutils literal notranslate"><span class="pre">WGCNA.CalculateSignedKME()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.CoexpressionModulePlot"><code class="docutils literal notranslate"><span class="pre">WGCNA.CoexpressionModulePlot()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.PPI_network"><code class="docutils literal notranslate"><span class="pre">WGCNA.PPI_network()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.TOMsimilarity"><code class="docutils literal notranslate"><span class="pre">WGCNA.TOMsimilarity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.adjacency"><code class="docutils literal notranslate"><span class="pre">WGCNA.adjacency()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.analyseWGCNA"><code class="docutils literal notranslate"><span class="pre">WGCNA.analyseWGCNA()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.barplotModuleEigenGene"><code class="docutils literal notranslate"><span class="pre">WGCNA.barplotModuleEigenGene()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.calBlockSize"><code class="docutils literal notranslate"><span class="pre">WGCNA.calBlockSize()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.checkAdjMat"><code class="docutils literal notranslate"><span class="pre">WGCNA.checkAdjMat()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.checkAndScaleWeights"><code class="docutils literal notranslate"><span class="pre">WGCNA.checkAndScaleWeights()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.checkSets"><code class="docutils literal notranslate"><span class="pre">WGCNA.checkSets()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.checkSimilarity"><code class="docutils literal notranslate"><span class="pre">WGCNA.checkSimilarity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.consensusMEDissimilarityMajor"><code class="docutils literal notranslate"><span class="pre">WGCNA.consensusMEDissimilarityMajor()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.consensusOrderMEs"><code class="docutils literal notranslate"><span class="pre">WGCNA.consensusOrderMEs()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.cutree"><code class="docutils literal notranslate"><span class="pre">WGCNA.cutree()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.cutreeHybrid"><code class="docutils literal notranslate"><span class="pre">WGCNA.cutreeHybrid()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.findModules"><code class="docutils literal notranslate"><span class="pre">WGCNA.findModules()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.fixDataStructure"><code class="docutils literal notranslate"><span class="pre">WGCNA.fixDataStructure()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.functional_enrichment_analysis"><code class="docutils literal notranslate"><span class="pre">WGCNA.functional_enrichment_analysis()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.getDatTraits"><code class="docutils literal notranslate"><span class="pre">WGCNA.getDatTraits()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.getGeneModule"><code class="docutils literal notranslate"><span class="pre">WGCNA.getGeneModule()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.getModuleName"><code class="docutils literal notranslate"><span class="pre">WGCNA.getModuleName()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.getModulesGene"><code class="docutils literal notranslate"><span class="pre">WGCNA.getModulesGene()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.goodGenesFun"><code class="docutils literal notranslate"><span class="pre">WGCNA.goodGenesFun()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.goodSamplesFun"><code class="docutils literal notranslate"><span class="pre">WGCNA.goodSamplesFun()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.goodSamplesGenes"><code class="docutils literal notranslate"><span class="pre">WGCNA.goodSamplesGenes()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.hclust"><code class="docutils literal notranslate"><span class="pre">WGCNA.hclust()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.intramodularConnectivity"><code class="docutils literal notranslate"><span class="pre">WGCNA.intramodularConnectivity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.labels2colors"><code class="docutils literal notranslate"><span class="pre">WGCNA.labels2colors()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.mergeCloseModules"><code class="docutils literal notranslate"><span class="pre">WGCNA.mergeCloseModules()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.moduleEigengenes"><code class="docutils literal notranslate"><span class="pre">WGCNA.moduleEigengenes()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.module_trait_relationships_heatmap"><code class="docutils literal notranslate"><span class="pre">WGCNA.module_trait_relationships_heatmap()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.multiSetMEs"><code class="docutils literal notranslate"><span class="pre">WGCNA.multiSetMEs()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.orderMEs"><code class="docutils literal notranslate"><span class="pre">WGCNA.orderMEs()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.pickSoftThreshold"><code class="docutils literal notranslate"><span class="pre">WGCNA.pickSoftThreshold()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.plotModuleEigenGene"><code class="docutils literal notranslate"><span class="pre">WGCNA.plotModuleEigenGene()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.preprocess"><code class="docutils literal notranslate"><span class="pre">WGCNA.preprocess()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.replaceMissing"><code class="docutils literal notranslate"><span class="pre">WGCNA.replaceMissing()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.request_PPI"><code class="docutils literal notranslate"><span class="pre">WGCNA.request_PPI()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.request_PPI_image"><code class="docutils literal notranslate"><span class="pre">WGCNA.request_PPI_image()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.request_PPI_subset"><code class="docutils literal notranslate"><span class="pre">WGCNA.request_PPI_subset()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.runWGCNA"><code class="docutils literal notranslate"><span class="pre">WGCNA.runWGCNA()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.saveWGCNA"><code class="docutils literal notranslate"><span class="pre">WGCNA.saveWGCNA()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.scaleFreeFitIndex"><code class="docutils literal notranslate"><span class="pre">WGCNA.scaleFreeFitIndex()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.setMetadataColor"><code class="docutils literal notranslate"><span class="pre">WGCNA.setMetadataColor()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.softConnectivity"><code class="docutils literal notranslate"><span class="pre">WGCNA.softConnectivity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.top_n_hub_genes"><code class="docutils literal notranslate"><span class="pre">WGCNA.top_n_hub_genes()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.updateGeneInfo"><code class="docutils literal notranslate"><span class="pre">WGCNA.updateGeneInfo()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.wgcna.WGCNA.updateSampleInfo"><code class="docutils literal notranslate"><span class="pre">WGCNA.updateSampleInfo()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison"><code class="docutils literal notranslate"><span class="pre">Comparison</span></code></a><ul>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.calculateFraction"><code class="docutils literal notranslate"><span class="pre">Comparison.calculateFraction()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.calculateJaccardSimilarity"><code class="docutils literal notranslate"><span class="pre">Comparison.calculateJaccardSimilarity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.calculatePvalue"><code class="docutils literal notranslate"><span class="pre">Comparison.calculatePvalue()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.compareNetworks"><code class="docutils literal notranslate"><span class="pre">Comparison.compareNetworks()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.jaccard"><code class="docutils literal notranslate"><span class="pre">Comparison.jaccard()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.plotBubbleComparison"><code class="docutils literal notranslate"><span class="pre">Comparison.plotBubbleComparison()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.plotHeatmapComparison"><code class="docutils literal notranslate"><span class="pre">Comparison.plotHeatmapComparison()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.plotJaccardSimilarity"><code class="docutils literal notranslate"><span class="pre">Comparison.plotJaccardSimilarity()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.comparison.Comparison.saveComparison"><code class="docutils literal notranslate"><span class="pre">Comparison.saveComparison()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#PyWGCNA.utils.compareNetworks"><code class="docutils literal notranslate"><span class="pre">compareNetworks()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.utils.compareSingleCell"><code class="docutils literal notranslate"><span class="pre">compareSingleCell()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.utils.getGeneList"><code class="docutils literal notranslate"><span class="pre">getGeneList()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.utils.getGeneListGOid"><code class="docutils literal notranslate"><span class="pre">getGeneListGOid()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.utils.readComparison"><code class="docutils literal notranslate"><span class="pre">readComparison()</span></code></a></li>
<li><a class="reference internal" href="#PyWGCNA.utils.readWGCNA"><code class="docutils literal notranslate"><span class="pre">readWGCNA()</span></code></a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="cite.html" title="Previous Chapter: Citation"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Citation</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/api.rst.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="body col-md-12 content" role="main">
      
  <section id="module-PyWGCNA.geneExp">
<span id="api-documentation"></span><h1>API Documentation<a class="headerlink" href="#module-PyWGCNA.geneExp" title="Link to this heading">¶</a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="PyWGCNA.geneExp.GeneExp">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">PyWGCNA.geneExp.</span></span><span class="sig-name descname"><span class="pre">GeneExp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">species</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'gene'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anndata</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneExp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneExpPath</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sampleInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/geneExp.html#GeneExp"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.geneExp.GeneExp" title="Link to this definition">¶</a></dt>
<dd><p>A class used to creat gene expression anndata along data trait including both genes and samples information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>species</strong> (<em>str</em>) – species of the data you use i.e mouse, human</p></li>
<li><p><strong>level</strong> (<em>str</em>) – which type of data you use including gene, transcript (default: gene)</p></li>
<li><p><strong>anndata</strong> – if the expression data is in anndata format you should pass it through this parameter. X should be expression matrix. var is a sample information and obs is a gene information.</p></li>
<li><p><strong>anndata</strong> – anndata</p></li>
<li><p><strong>geneExp</strong> (<em>pandas dataframe</em>) – expression matrix which genes are in the rows and samples are columns</p></li>
<li><p><strong>geneExpPath</strong> (<em>str</em>) – path of expression matrix</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – separation symbol to use for reading data in geneExpPath properly</p></li>
<li><p><strong>geneInfo</strong> (<em>pandas dataframe</em>) – dataframe that contains genes information it should have a same index as gene expression column names (gene/transcript ID)</p></li>
<li><p><strong>sampleInfo</strong> (<em>pandas dataframe</em>) – dataframe that contains samples information it should have a same index as gene expression index (sample ID)</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.geneExp.GeneExp.updateGeneInfo">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">updateGeneInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">geneExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/geneExp.html#GeneExp.updateGeneInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.geneExp.GeneExp.updateGeneInfo" title="Link to this definition">¶</a></dt>
<dd><p>add/update genes info in expr anndata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>geneExpr</strong> (<em>anndata</em>) – gene expression data along with sample and genes/transcript information</p></li>
<li><p><strong>geneInfo</strong> (<em>pandas dataframe</em>) – gene information table you want to add to your data</p></li>
<li><p><strong>path</strong> (<em>str</em>) – path of geneInfo</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – separation symbol to use for reading data in path properly (default: ‘,’)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>updated gene expression data along with sample and genes/transcript information</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>anndata</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.geneExp.GeneExp.updateSampleInfo">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">updateSampleInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">geneExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sampleInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/geneExp.html#GeneExp.updateSampleInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.geneExp.GeneExp.updateSampleInfo" title="Link to this definition">¶</a></dt>
<dd><p>add/update metadata in expr anndata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>geneExpr</strong> (<em>anndata</em>) – gene expression data along with sample and genes/transcript information</p></li>
<li><p><strong>sampleInfo</strong> (<em>pandas dataframe</em>) – Sample information table you want to add to your data</p></li>
<li><p><strong>path</strong> (<em>str</em>) – path of metaData</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – separation symbol to use for reading data in path properly (default: ‘,’)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>updated gene expression data along with sample and genes/transcript information</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>anndata</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class" id="module-PyWGCNA.wgcna">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">PyWGCNA.wgcna.</span></span><span class="sig-name descname"><span class="pre">WGCNA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'WGCNA'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TPMcutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">powers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">RsquaredCut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.9</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">MeanCut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">networkType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'signed</span> <span class="pre">hybrid'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TOMType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'signed'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minModuleSize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">naColor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'grey'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">inf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">MEDissThres</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'gene'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anndata</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneExp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneExpPath</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outputPath</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figureType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA" title="Link to this definition">¶</a></dt>
<dd><p>A class used to do weighted gene co-expression network analysis.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> (<em>str</em>) – name of the WGCNA we used to visualize data (default: ‘WGCNA’)</p></li>
<li><p><strong>save</strong> (<em>bool</em>) – indicate if you want to save result of important steps in a figure directory (default: False)</p></li>
<li><p><strong>outputPath</strong> (<em>str</em>) – path you want to save all you figures and object (default: ‘’, where you rau your script)</p></li>
<li><p><strong>geneExp</strong> (<em>geneExp class</em>) – expression matrix</p></li>
<li><p><strong>datExpr</strong> (<em>anndata</em>) – data expression data that contains preprocessed data</p></li>
<li><p><strong>TPMcutoff</strong> (<em>int</em>) – cut off for removing genes that expressed under this number along samples</p></li>
<li><p><strong>cut</strong> (<em>float</em>) – number to remove outlier sample (default: ‘inf’) By default we don’t remove any sample by hierarchical clustering</p></li>
<li><p><strong>powers</strong> (<em>list</em><em> of </em><em>int</em>) – different powers to test to have scale free network (default: [1:10, 11:21:2])</p></li>
<li><p><strong>RsquaredCut</strong> (<em>float</em>) – R squaered cut to choose power for having scale free network; between 0 to 1 (default: 0.9)</p></li>
<li><p><strong>MeanCut</strong> (<em>int</em>) – mean connectivity to choose power for having scale free network (default: 100)</p></li>
<li><p><strong>power</strong> (<em>int</em>) – power to have scale free network (default: 6)</p></li>
<li><p><strong>sft</strong> (<em>pandas dataframe</em>) – soft threshold table which has information for each powers</p></li>
<li><p><strong>networkType</strong> (<em>str</em>) – Type of network we can create including “unsigned”, “signed” and “signed hybrid” (default: “signed hybrid”)</p></li>
<li><p><strong>adjacency</strong> (<em>ndarray</em>) – adjacency matrix calculating base of the type of network</p></li>
<li><p><strong>geneTree</strong> (<em>ndarray</em>) – average hierarchical clustering of dissTOM matrix</p></li>
<li><p><strong>TOMType</strong> (<em>str</em>) – Type of topological overlap matrix(TOM) including “unsigned”, “signed” (default: “signed”)</p></li>
<li><p><strong>TOM</strong> – topological overlap measure using average linkage hierarchical clustering which inputs a measure of interconnectedness</p></li>
<li><p><strong>TOM</strong> – ndarray</p></li>
<li><p><strong>minModuleSize</strong> (<em>int</em>) – We like large modules, so we set the minimum module size relatively high (default: 50)</p></li>
<li><p><strong>dynamicMods</strong> (<em>list</em>) – name of modules by clustering similar genes together</p></li>
<li><p><strong>naColor</strong> (<em>str</em>) – color we used to identify genes we don’t find any cluster for them (default: “grey”)</p></li>
<li><p><strong>MEs</strong> (<em>ndarray</em>) – eigengenes</p></li>
<li><p><strong>MEDissThres</strong> (<em>float</em>) – diss similarity threshold (default: 0.2)</p></li>
<li><p><strong>datME</strong> (<em>pandas dataframe</em>) – </p></li>
</ul>
</dd>
</dl>
<p>:param signedKME:(signed) eigengene-based connectivity (module membership)
:type signedKME: pandas dataframe
:param moduleTraitCor: correlation between each module and metadata
:type moduleTraitCor: pandas dataframe
:param moduleTraitPvalue: p-value of correlation between each module and metadata
:type moduleTraitPvalue: pandas dataframe
:param figureType: extension of figure (default: “pdf”)</p>
<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.CalculateSignedKME">
<span class="sig-name descname"><span class="pre">CalculateSignedKME</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exprWeights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">MEWeights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.CalculateSignedKME"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.CalculateSignedKME" title="Link to this definition">¶</a></dt>
<dd><p>Calculation of (signed) eigengene-based connectivity, also known as module membership.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exprWeights</strong> (<em>pandas dataframe</em>) – optional weight matrix of observation weights for datExpr, of the same dimensions as datExpr</p></li>
<li><p><strong>MEWeights</strong> (<em>pandas dataframe</em>) – optional weight matrix of observation weights for datME, of the same dimensions as datME</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A data frame in which rows correspond to input genes and columns to module eigengenes, giving the signed eigengene-based connectivity of each gene with respect to each eigengene.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.CoexpressionModulePlot">
<span class="sig-name descname"><span class="pre">CoexpressionModulePlot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">modules</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">numGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">numConnections</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minTOM</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filters</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.CoexpressionModulePlot"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.CoexpressionModulePlot" title="Link to this definition">¶</a></dt>
<dd><p>plot Coexpression for given module</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>modules</strong> (<em>list</em><em> of </em><em>str</em>) – name of modules you like to plot</p></li>
<li><p><strong>numGenes</strong> (<em>int</em>) – number of genes you want to show for each module</p></li>
<li><p><strong>numConnections</strong> (<em>int</em>) – number of connection you want to show for each module</p></li>
<li><p><strong>minTOM</strong> (<em>float</em>) – minimum TOM to keep connections</p></li>
<li><p><strong>filters</strong> (<em>dict</em>) – Dictionary which keys are columns names of datExpr.var that you want to filter the genes based on it and values are determining which rows you want to keep</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name of the html output file (default: module names or network if there is more than 3 modules for input)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>save a html file with name of modules in figures directory</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.PPI_network">
<span class="sig-name descname"><span class="pre">PPI_network</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">species</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">moduleName</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneList</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'image'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.PPI_network"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.PPI_network" title="Link to this definition">¶</a></dt>
<dd><p>retrieve an image of a STRING network of a neighborhood surrounding one or more proteins or ask STRING to show only the network of interactions between your input proteins.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>species</strong> (<em>int</em>) – NCBI taxon identifiers (e.g. Human is 9606, see: <a class="reference external" href="https://string-db.org/cgi/input.pl?input_page_active_form=organisms">https://string-db.org/cgi/input.pl?input_page_active_form=organisms</a>).</p></li>
<li><p><strong>moduleName</strong> (<em>str</em>) – name of module you want to find PPI</p></li>
<li><p><strong>geneList</strong> (<em>list</em>) – list of genes you want to find PPI</p></li>
<li><p><strong>output_format</strong> (<em>str</em>) – format of output which can be “image”, “highres_image”, “svg” (default: “image”)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>dataframe contains genes along with interaction with their scores</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.TOMsimilarity">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">TOMsimilarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adjMat</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TOMType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'signed'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TOMDenom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'min'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.TOMsimilarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.TOMsimilarity" title="Link to this definition">¶</a></dt>
<dd><p>Calculation of the topological overlap matrix, and the corresponding dissimilarity, from a given adjacency matrix</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adjMat</strong> (<em>pandas dataframe</em>) – adjacency matrix, that is a square, symmetric matrix with entries between 0 and 1 (negative values are allowed if TOMType==”signed”).</p></li>
<li><p><strong>TOMType</strong> (<em>str</em>) – one of “unsigned”, “signed”</p></li>
<li><p><strong>TOMDenom</strong> (<em>str</em>) – a character string specifying the TOM variant to be used. Recognized values are “min” giving the standard TOM described in Zhang and Horvath (2005), and “mean” in which the min function in the denominator is replaced by mean. The “mean” may produce better results but at this time should be considered experimental.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A matrix holding the topological overlap.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.adjacency">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">adjacency</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selectCols=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">adjacencyType='unsigned'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">power=6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">corOptions=Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weightArgNames=None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.adjacency"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.adjacency" title="Link to this definition">¶</a></dt>
<dd><p>Calculates (correlation or distance) network adjacency from given expression data or from a similarity</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>datExpr</strong> (<em>pandas dataframe</em>) – data frame containing expression data. Columns correspond to genes and rows to samples.</p></li>
<li><p><strong>selectCols</strong> (<em>list</em>) – for correlation networks only; can be used to select genes whose adjacencies will be calculated. Should be either a numeric list giving the indices of the genes to be used, or a boolean list indicating which genes are to be used.</p></li>
<li><p><strong>adjacencyType</strong> (<em>str</em>) – adjacency network type. Allowed values are (unique abbreviations of) “unsigned”, “signed”, “signed hybrid”. (default = unsigned)</p></li>
<li><p><strong>power</strong> (<em>int</em>) – soft thresholding power.</p></li>
<li><p><strong>corOptions</strong> (<em>pandas dataframe</em>) – specifying additional arguments to be passed to the function given by corFnc.</p></li>
<li><p><strong>weights</strong> (<em>pandas dataframe</em>) – optional observation weights for datExpr to be used in correlation calculation. A matrix of the same dimensions as datExpr, containing non-negative weights. Only used with Pearson correlation.</p></li>
<li><p><strong>weightArgNames</strong> (<em>list</em>) – character list of length 2 giving the names of the arguments to corFnc that represent weights for variable x and y. Only used if weights are non-NULL.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Adjacency matrix</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.analyseWGCNA">
<span class="sig-name descname"><span class="pre">analyseWGCNA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">order</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneList</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.analyseWGCNA"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.analyseWGCNA" title="Link to this definition">¶</a></dt>
<dd><p>Analysing results: 1.calculating module trait relationship 2.plotting module heatmap eigengene 3.finding GO term for each module</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>order</strong> (<em>list</em>) – indicate in which order metadata will show up in plots (should same as metadata name in anndata)</p></li>
<li><p><strong>geneList</strong> (<em>pandas dataframe</em>) – genes information you want to add (keep in mind you can not have multiple row for same gene)</p></li>
<li><p><strong>show</strong> (<em>bool</em>) – indicate if you want to see plots in when you run your code</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.barplotModuleEigenGene">
<span class="sig-name descname"><span class="pre">barplotModuleEigenGene</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">moduleName</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metadata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">combine</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorBar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.barplotModuleEigenGene"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.barplotModuleEigenGene" title="Link to this definition">¶</a></dt>
<dd><p>bar plot of module eigen gene figure in given module</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>moduleName</strong> (<em>str</em>) – module name</p></li>
<li><p><strong>metadata</strong> (<em>list</em>) – list of metadata you want to be plotted</p></li>
<li><p><strong>combine</strong> (<em>bool</em>) – indicate if you want to combine all metadata to show them together</p></li>
<li><p><strong>show</strong> (<em>bool</em>) – indicate if you want to see plots in when you run your code</p></li>
</ul>
</dd>
<dt class="field-even">Praram colorBar<span class="colon">:</span></dt>
<dd class="field-even"><p>metadata you want to use to color bar plot with</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.calBlockSize">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">calBlockSize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">matrixSize</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rectangularBlocks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxMemoryAllocation</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overheadFactor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.calBlockSize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.calBlockSize" title="Link to this definition">¶</a></dt>
<dd><p>find suitable block size for calculating soft power threshold</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.checkAdjMat">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">checkAdjMat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adjMat</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.checkAdjMat"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.checkAdjMat" title="Link to this definition">¶</a></dt>
<dd><p>check adjacency matrix format is correct</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adjMat</strong> (<em>pandas dataframe</em>) – data we want to be checked</p></li>
<li><p><strong>min</strong> (<em>int</em>) – minimum value to be allowed for data (default = 0)</p></li>
<li><p><strong>max</strong> (<em>int</em>) – maximum value to be allowed for data (default = 1)</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>exit</strong> – if format is not correct</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.checkAndScaleWeights">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">checkAndScaleWeights</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">weights</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">expr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scaleByMax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.checkAndScaleWeights"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.checkAndScaleWeights" title="Link to this definition">¶</a></dt>
<dd><p>check and scale weights of gene expression
:param weights: weights of gene expression
:type weights: pandas dataframe
:param expr: gene expression matrix
:type expr: pandas dataframe
:param scaleByMax: if you want to scale your weights by diving to max
:type scaleByMax: boll</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>processed weights of gene expression</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.checkSets">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">checkSets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">checkStructure</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.checkSets"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.checkSets" title="Link to this definition">¶</a></dt>
<dd><p>Checks whether given sets have the correct format and retrieves dimensions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> (<em>dict</em>) – A dict of lists; in each list there must be a component named data whose content is a matrix or dataframe or array of dimension 2.</p></li>
<li><p><strong>checkStructure</strong> (<em>bool</em>) – If FALSE, incorrect structure of data will trigger an error. If TRUE, an appropriate flag (see output) will be set to indicate whether data has correct structure. (default = False)</p></li>
<li><p><strong>useSets</strong> (<em>list</em>) – Optional specification of entries of the list data that are to be checked. Defaults to all components. This may be useful when data only contains information for some of the sets.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>a dictionary contains: “nSets”: Number of sets (length of the vector data). “nGenes”: Number of columns in the data components in the lists. This number must be the same for all sets. “nSamples”: A vector of length nSets giving the number of rows in the data components. “structureOK”: Only set if the argument checkStructure equals TRUE. The value is TRUE if the paramter data passes a few tests of its structure, and FALSE otherwise. The tests are not exhaustive and are meant to catch obvious user errors rather than be bulletproof.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.checkSimilarity">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">checkSimilarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adjMat</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.checkSimilarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.checkSimilarity" title="Link to this definition">¶</a></dt>
<dd><p>check similarity matrix format is correct</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adjMat</strong> (<em>pandas dataframe</em>) – data we want to be checked</p></li>
<li><p><strong>min</strong> (<em>int</em>) – minimum value to be allowed for data (default = 0)</p></li>
<li><p><strong>max</strong> (<em>int</em>) – maximum value to be allowed for data (default = 1)</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>exit</strong> – if format is not correct</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.consensusMEDissimilarityMajor">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">consensusMEDissimilarityMajor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">MEs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useAbs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'consensus'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.consensusMEDissimilarityMajor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.consensusMEDissimilarityMajor" title="Link to this definition">¶</a></dt>
<dd><p>Calculates consensus dissimilarity (1-cor) of given module eigengenes realized in several sets.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.consensusOrderMEs">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">consensusOrderMEs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">MEs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useAbs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">greyLast</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">greyName</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'MEgrey'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'consensus'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.consensusOrderMEs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.consensusOrderMEs" title="Link to this definition">¶</a></dt>
<dd><p>Reorder given (eigen-)vectors such that similar ones (as measured by correlation) are next to each other.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>MEs</strong> (<em>dict</em>) – Module eigengenes of several sets in a multi-set format</p></li>
<li><p><strong>useAbs</strong> (<em>bool</em>) – Controls whether vector similarity should be given by absolute value of correlation or plain correlation. (defualt = False)</p></li>
<li><p><strong>useSet</strong> – Allows the user to specify for which sets the eigengene ordering is to be performed.</p></li>
<li><p><strong>greyLast</strong> (<em>bool</em>) – Normally the color grey is reserved for unassigned genes; hence the grey module is not a proper module and it is conventional to put it last. If this is not desired, set the parameter to FALSE. (defualt = True)</p></li>
<li><p><strong>greyName</strong> (<em>str</em>) – Name of the grey module eigengene. (defualt = “MEgrey”)</p></li>
<li><p><strong>method</strong> (<em>str</em>) – A character string giving the method to be used calculating the consensus dissimilarity. Allowed values are (abbreviations of) “consensus” and “majority”. The consensus dissimilarity is calculated as the maximum of given set dissimilarities for “consensus” and as the average for “majority”.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary of the same type as MEs containing the re-ordered eigengenes</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.cutree">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cutree</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sampleTree</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutHeight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50000.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.cutree"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.cutree" title="Link to this definition">¶</a></dt>
<dd><p>Given a linkage matrix Z, return the cut tree. remove samples/genes/modules base on hierarchical clustering</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sampleTree</strong> (<em>scipy.cluster.linkage array</em>) – The linkage matrix.</p></li>
<li><p><strong>cutHeight</strong> (<em>array_like</em>) – A optional height at which to cut the tree (default = 50000)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>An array indicating group membership at each agglomeration step. I.e., for a full cut tree, in the first column each data point is in its own cluster. At the next step, two nodes are merged. Finally, all singleton and non-singleton clusters are in one group. If n_clusters or height are given, the columns correspond to the columns of n_clusters or height.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>array</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.cutreeHybrid">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cutreeHybrid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dendro</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distM</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutHeight=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minClusterSize=20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">deepSplit=1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxCoreScatter=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minGap=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxAbsCoreScatter=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minAbsGap=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minSplitHeight=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minAbsSplitHeight=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">externalBranchSplitFnc=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nExternalSplits=0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minExternalSplit=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">externalSplitOptions=Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">externalSplitFncNeedsDistance=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assumeSimpleExternalSpecification=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pamStage=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pamRespectsDendro=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useMedoids=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxPamDist=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">respectSmallClusters=True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.cutreeHybrid"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.cutreeHybrid" title="Link to this definition">¶</a></dt>
<dd><p>Detect clusters in a dendorgram produced by the function hclust.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dendro</strong> (<em>ndarray</em>) – a hierarchical clustering dendorgram such as one returned by hclust.</p></li>
<li><p><strong>distM</strong> (<em>pandas dataframe</em>) – Distance matrix that was used as input to hclust.</p></li>
<li><p><strong>cutHeight</strong> (<em>int</em>) – Maximum joining heights that will be considered. It defaults to 99of the range between the 5th percentile and the maximum of the joining heights on the dendrogram.</p></li>
<li><p><strong>minClusterSize</strong> (<em>int</em>) – Minimum cluster size. (default = 20)</p></li>
<li><p><strong>deepSplit</strong> (<em>int</em><em> or </em><em>bool</em>) – Either logical or integer in the range 0 to 4. Provides a rough control over sensitivity to cluster splitting. The higher the value, the more and smaller clusters will be produced. (default = 1)</p></li>
<li><p><strong>maxCoreScatter</strong> (<em>int</em>) – Maximum scatter of the core for a branch to be a cluster, given as the fraction of cutHeight relative to the 5th percentile of joining heights.</p></li>
<li><p><strong>minGap</strong> (<em>int</em>) – Minimum cluster gap given as the fraction of the difference between cutHeight and the 5th percentile of joining heights.</p></li>
<li><p><strong>maxAbsCoreScatter</strong> (<em>int</em>) – Maximum scatter of the core for a branch to be a cluster given as absolute heights. If given, overrides maxCoreScatter.</p></li>
<li><p><strong>minAbsGap</strong> (<em>int</em>) – Minimum cluster gap given as absolute height difference. If given, overrides minGap.</p></li>
<li><p><strong>minSplitHeight</strong> (<em>int</em>) – Minimum split height given as the fraction of the difference between cutHeight and the 5th percentile of joining heights. Branches merging below this height will automatically be merged. Defaults to zero but is used only if minAbsSplitH</p></li>
<li><p><strong>minAbsSplitHeight</strong> (<em>int</em>) – Minimum split height given as an absolute height. Branches merging below this height will automatically be merged. If not given (default), will be determined from minSplitHeight above.</p></li>
<li><p><strong>externalBranchSplitFnc</strong> – Optional function to evaluate split (dissimilarity) between two branches. Either a single function or a list in which each component is a function.</p></li>
<li><p><strong>minExternalSplit</strong> (<em>list</em>) – Thresholds to decide whether two branches should be merged. It should be a numeric list of the same length as the number of functions in externalBranchSplitFnc above.</p></li>
<li><p><strong>externalSplitOptions</strong> (<em>pandas dataframe</em>) – Further arguments to function externalBranchSplitFnc. If only one external function is specified in externalBranchSplitFnc above, externalSplitOptions can be a named list of arguments or a list with one component.</p></li>
<li><p><strong>externalSplitFncNeedsDistance</strong> (<em>pandas dataframe</em>) – Optional specification of whether the external branch split functions need the distance matrix as one of their arguments. Either NULL or a logical list with one element per branch</p></li>
<li><p><strong>assumeSimpleExternalSpecification</strong> (<em>bool</em>) – when minExternalSplit above is a scalar (has length 1), should the function assume a simple specification of externalBranchSplitFnc and externalSplitOptions. (default = True)</p></li>
<li><p><strong>pamStage</strong> (<em>bool</em>) – If TRUE, the second (PAM-like) stage will be performed. (default = True)</p></li>
<li><p><strong>pamRespectsDendro</strong> (<em>bool</em>) – If TRUE, the PAM stage will respect the dendrogram in the sense an object can be PAM-assigned only to clusters that lie below it on the branch that the object is merged into. (default = True)</p></li>
<li><p><strong>useMedoids</strong> – if TRUE, the second stage will be use object to medoid distance; if FALSE, it will use average object to cluster distance. (default = False)</p></li>
<li><p><strong>maxPamDist</strong> (<em>float</em>) – Maximum object distance to closest cluster that will result in the object assigned to that cluster. Defaults to cutHeight.</p></li>
<li><p><strong>respectSmallClusters</strong> (<em>bool</em>) – If TRUE, branches that failed to be clusters in stage 1 only because of insufficient size will be assigned together in stage 2. If FALSE, all objects will be assigned individually. (default = False)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>list detailing the deteced branch structure.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.findModules">
<span class="sig-name descname"><span class="pre">findModules</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.findModules"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.findModules" title="Link to this definition">¶</a></dt>
<dd><p>Clustering genes through original WGCNA pipeline: 1.pick soft threshold 2.calculating adjacency matrix 3.calculating TOM similarity matrix 4.cluster genes base of dissTOM 5.merge similar cluster dynamically</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.fixDataStructure">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">fixDataStructure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.fixDataStructure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.fixDataStructure" title="Link to this definition">¶</a></dt>
<dd><p>Encapsulates single-set data in a wrapper that makes the data suitable for functions working on multiset data collections.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>data</strong> (<em>pandas dataframe ot dict</em>) – A dataframe, matrix or array with two dimensions to be encapsulated.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>input data in a format suitable for functions operating on multiset data collections.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.functional_enrichment_analysis">
<span class="sig-name descname"><span class="pre">functional_enrichment_analysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">moduleName</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_value</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.functional_enrichment_analysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.functional_enrichment_analysis" title="Link to this definition">¶</a></dt>
<dd><p>Doing functional enrichment analysis including GO, KEGG and REACTOME</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>type</strong> (<em>str</em>) – indicate the type of databases which it should be one of “GO”, “KEGG”, “REACTOME”</p></li>
<li><p><strong>moduleName</strong> (<em>str</em>) – module name</p></li>
<li><p><strong>sets</strong> (<em>str</em><em>, </em><em>list</em><em>, </em><em>tuple</em>) – str, list, tuple of Enrichr Library name(s). or custom defined gene_sets (dict, or gmt file) (you can add any Enrichr Libraries from here: <a class="reference external" href="https://maayanlab.cloud/Enrichr/#stats">https://maayanlab.cloud/Enrichr/#stats</a>) only need to fill if the type is GO or KEGG</p></li>
<li><p><strong>p_value</strong> (<em>float</em>) – Defines the pValue threshold. (default: 0.05)</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name of the file you want to use to save plot (default is moduleName)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.getDatTraits">
<span class="sig-name descname"><span class="pre">getDatTraits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">metaData</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.getDatTraits"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.getDatTraits" title="Link to this definition">¶</a></dt>
<dd><p>get data trait module base on samples information</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>a dataframe contains information in suitable format for plotting module trait relationship heatmap</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.getGeneModule">
<span class="sig-name descname"><span class="pre">getGeneModule</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">moduleName</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.getGeneModule"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.getGeneModule" title="Link to this definition">¶</a></dt>
<dd><p>get list of genes corresponding to modules</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>moduleName</strong> (<em>list</em>) – name of modules</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary contains list of genes for requested module(s)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.getModuleName">
<span class="sig-name descname"><span class="pre">getModuleName</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.getModuleName"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.getModuleName" title="Link to this definition">¶</a></dt>
<dd><p>get names of modules</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>name of modules</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.getModulesGene">
<span class="sig-name descname"><span class="pre">getModulesGene</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">geneIds</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.getModulesGene"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.getModulesGene" title="Link to this definition">¶</a></dt>
<dd><p>get list of modules corresponding to gene(s)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>geneIds</strong> (<em>list</em><em> or </em><em>str</em>) – gene id</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A list contains name of module(s) for requested gene(s)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list or str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.goodGenesFun">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">goodGenesFun</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSamples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minFraction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNSamples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minRelativeWeight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.goodGenesFun"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.goodGenesFun" title="Link to this definition">¶</a></dt>
<dd><p>Check data for missing entries and returns a list of genes that have non-zero variance</p>
<p>:param datExpr:expression data. A data frame in which columns are genes and rows ar samples.
:type datExpr: pandas dataframe
:param weights: optional observation weights in the same format (and dimensions) as datExpr.
:type weights: pandas dataframe
:param useSamples: optional specifications of which samples to use for the check (Defaults to using all samples)
:type useSamples: list of bool
:param useGenes: optional specifications of genes for which to perform the check (Defaults to using all genes)
:type useGenes: list of bool
:param minFraction: minimum fraction of non-missing samples for a gene to be considered good. (default = 1/2)
:type minFraction: float
:param minNSamples: minimum number of non-missing samples for a gene to be considered good. (default = 4)
:type minNSamples: int
:param minNGenes: minimum number of good genes for the data set to be considered fit for analysis. If the actual number of good genes falls below this threshold, an error will be issued. (default = 4)
:type minNGenes: int
:param tol: An optional ‘small’ number to compare the variance against
:type tol: float
:param minRelativeWeight: observations whose relative weight is below this threshold will be considered missing. Here relative weight is weight divided by the maximum weight in the column (gene). (default = 0.1)
:type minRelativeWeight: float</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A logical list with one entry per gene that is TRUE if the gene is considered good and FALSE otherwise. Note that all genes excluded by useGenes are automatically assigned FALSE.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>list of bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.goodSamplesFun">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">goodSamplesFun</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSamples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minFraction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNSamples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minRelativeWeight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.goodSamplesFun"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.goodSamplesFun" title="Link to this definition">¶</a></dt>
<dd><p>Check data for missing entries and returns a list of samples that have non-zero variance</p>
<p>:param datExpr:expression data. A data frame in which columns are genes and rows ar samples.
:type datExpr: pandas dataframe
:param weights: optional observation weights in the same format (and dimensions) as datExpr.
:type weights: pandas dataframe
:param useSamples: optional specifications of which samples to use for the check (Defaults to using all samples)
:type useSamples: list of bool
:param useGenes: optional specifications of genes for which to perform the check (Defaults to using all genes)
:type useGenes: list of bool
:param minFraction: minimum fraction of non-missing samples for a gene to be considered good. (default = 1/2)
:type minFraction: float
:param minNSamples: findModulesminimum number of non-missing samples for a gene to be considered good. (default = 4)
:type minNSamples: int
:param minNGenes: minimum number of good genes for the data set to be considered fit for analysis. If the actual number of good genes falls below this threshold, an error will be issued. (default = 4)
:type minNGenes: int
:param minRelativeWeight: observations whose relative weight is below this threshold will be considered missing. Here relative weight is weight divided by the maximum weight in the column (gene). (default = 0.1)
:type minRelativeWeight: float</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A logical list with one entry per sample that is TRUE if the sample is considered good and FALSE otherwise. Note that all samples excluded by useSamples are automatically assigned FALSE.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>list of bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.goodSamplesGenes">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">goodSamplesGenes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minFraction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNSamples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minRelativeWeight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.goodSamplesGenes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.goodSamplesGenes" title="Link to this definition">¶</a></dt>
<dd><p>Checks data for missing entries, entries with weights below a threshold, and zero-variance genes. If necessary, the filtering is iterated.</p>
<p>:param datExpr:expression data. A data frame in which columns are genes and rows ar samples.
:type datExpr: pandas dataframe
:param weights: optional observation weights in the same format (and dimensions) as datExpr.
:type weights: pandas dataframe
:param minFraction: minimum fraction of non-missing samples for a gene to be considered good. (default = 1/2)
:type minFraction: float
:param minNSamples: minimum number of non-missing samples for a gene to be considered good. (default = 4)
:type minNSamples: int
:param minNGenes: minimum number of good genes for the data set to be considered fit for analysis. If the actual number of good genes falls below this threshold, an error will be issued. (default = 4)
:type minNGenes: int
:param tol: An optional ‘small’ number to compare the variance against
:type tol: float
:param minRelativeWeight: observations whose relative weight is below this threshold will be considered missing. Here relative weight is weight divided by the maximum weight in the column (gene). (default = 0.1)
:type minRelativeWeight: float</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A triple containing (goodGenes, goodSamples, allOK) goodSamples: A logical vector with one entry per sample that is TRUE if the sample is considered good and FALSE otherwise. goodGenes: A logical vector with one entry per gene that is TRUE if the gene is considered good and FALSE otherwise. allOK: if everything is okay</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>list, list, bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.hclust">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">hclust</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'complete'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.hclust"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.hclust" title="Link to this definition">¶</a></dt>
<dd><p>Hierarchical cluster analysis on a set of dissimilarities and methods for analyzing it.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>d</strong> (<em>ndarray</em>) – a dissimilarity structure as produced by ‘pdist’.</p></li>
<li><p><strong>method</strong> (<em>str</em>) – The linkage algorithm to use. (default = complete)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The hierarchical clustering encoded as a linkage matrix.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.intramodularConnectivity">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">intramodularConnectivity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mat</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scaleByMax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.intramodularConnectivity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.intramodularConnectivity" title="Link to this definition">¶</a></dt>
<dd><p>Calculates intramodular connectivity, i.e., connectivity of nodes to other nodes within the same module.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mat</strong> (<em>ndarray</em>) – adjacency which should be a square, symmetric matrix with entries between 0 and 1.</p></li>
<li><p><strong>colors</strong> (<em>list</em>) – module labels. A list of length ncol(adjMat) giving a module label for each gene (node) of the network.</p></li>
<li><p><strong>scaleByMax</strong> (<em>bool</em>) – should intramodular connectivity be scaled by the maximum IM connectivity in each module?</p></li>
<li><p><strong>index</strong> (<em>ndarray</em>) – gene id or name of mat index</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>If input getWholeNetworkConnectivity is TRUE, a data frame with 4 columns giving the total connectivity, intramodular connectivity, extra-modular connectivity, and the difference of the intra- and extra-modular connectivities for all genes; otherwise a vector of intramodular connectivities</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.labels2colors">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">labels2colors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zeroIsGrey</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorSeq</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">naColor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'grey'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.labels2colors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.labels2colors" title="Link to this definition">¶</a></dt>
<dd><p>Converts a vector or array of numerical labels into a corresponding vector or array of colors corresponding to the labels.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>labels</strong> (<em>list</em><em> or </em><em>matrix</em>) – list or matrix of non-negative integer or other (such as character) labels.</p></li>
<li><p><strong>zeroIsGrey</strong> (<em>bool</em>) – If TRUE, labels 0 will be assigned color grey. Otherwise, labels below 1 will trigger an error. (default = True)</p></li>
<li><p><strong>colorSeq</strong> (<em>list</em><em> or </em><em>matrix</em>) – Color sequence corresponding to labels. If not given, a standard sequence will be used.</p></li>
<li><p><strong>naColor</strong> (<em>str</em>) – Color that will encode missing values.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>An array of character strings of the same length or dimensions as labels.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.mergeCloseModules">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mergeCloseModules</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exprData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">MEs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">impute</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">checkDataFormat</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unassdColor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'grey'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useAbs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">equalizeQuantiles</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">quantileSummary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'mean'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">consensusQuantile</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutHeight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">iterate</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relabel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorSeq</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">getNewMEs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">getNewUnassdME</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trapErrors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.mergeCloseModules"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.mergeCloseModules" title="Link to this definition">¶</a></dt>
<dd><p>Merges modules in gene expression networks that are too close as measured by the correlation of their eigengenes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exprData</strong> (<em>pandas dataframe</em>) – Expression data, either a single data frame with rows corresponding to samples and columns to genes, or in a multi-set format.</p></li>
<li><p><strong>colors</strong> (<em>list</em>) – A list (numeric, character or a factor) giving module colors for genes. The method only makes sense when genes have the same color label in all sets, hence a single list.</p></li>
<li><p><strong>MEs</strong> (<em>dict</em>) – If module eigengenes have been calculated before, the user can save some computational time by inputting them. MEs should have the same format as exprData. If they are not given, they will be calculated.</p></li>
<li><p><strong>useSets</strong> (<em>list</em>) – A list of scalar allowing the user to specify which sets will be used to calculate the consensus dissimilarity of module eigengenes. Defaults to all given sets.</p></li>
<li><p><strong>impute</strong> (<em>bool</em>) – Should missing values be imputed in eigengene calculation? If imputation is disabled, the presence of NA entries will cause the eigengene calculation to fail and eigengenes will be replaced by their hubgene approximation. (defualt = True)</p></li>
<li><p><strong>checkDataFormat</strong> (<em>bool</em>) – If TRUE, the function will check exprData and MEs for correct multi-set structure. If single set data is given, it will be converted into a format usable for the function. If FALSE, incorrect structure of input data will trigger an error. (defualt = True)</p></li>
<li><p><strong>unassdColor</strong> (<em>str</em>) – Specifies the string that labels unassigned genes. Module of this color will not enter the module eigengene clustering and will not be merged with other modules. (default = “grey”)</p></li>
<li><p><strong>useAbs</strong> (<em>bool</em>) – Specifies whether absolute value of correlation or plain correlation (of module eigengenes) should be used in calculating module dissimilarity. (defualt = False)</p></li>
<li><p><strong>equalizeQuantiles</strong> (<em>bool</em>) – should quantiles of the eigengene dissimilarity matrix be equalized (“quantile normalized”)? The default is FALSE for reproducibility of old code; when there are many eigengenes (e.g., at least 50), better results may be achieved if quantile equalization is used. (defualt = False)</p></li>
<li><p><strong>quantileSummary</strong> (<em>str</em>) – One of “mean” or “median”. Controls how a reference dissimilarity is computed from the input ones (using mean or median, respectively). (default = “mean”)</p></li>
<li><p><strong>consensusQuantile</strong> (<em>int</em>) – A number giving the desired quantile to use in the consensus similarity calculation. (defualt = 0)</p></li>
<li><p><strong>cutHeight</strong> (<em>float</em>) – Maximum dissimilarity (i.e., 1-correlation) that qualifies modules for merging. (defualt = 0.2)</p></li>
<li><p><strong>iterate</strong> (<em>bool</em>) – Controls whether the merging procedure should be repeated until there is no change. If FALSE, only one iteration will be executed. (defualt = True)</p></li>
<li><p><strong>relabel</strong> (<em>bool</em>) – Controls whether, after merging, color labels should be ordered by module size. (defualt = False)</p></li>
<li><p><strong>colorSeq</strong> (<em>list</em>) – Color labels to be used for relabeling. Defaults to the standard color order used in this package if colors are not numeric, and to integers starting from 1 if colors is numeric.</p></li>
<li><p><strong>getNewMEs</strong> (<em>bool</em>) – Controls whether module eigengenes of merged modules should be calculated and returned. (defualt = True)</p></li>
<li><p><strong>getNewUnassdME</strong> (<em>bool</em>) – When doing module eigengene manipulations, the function does not normally calculate the eigengene of the ‘module’ of unassigned (‘grey’) genes. Setting this option to TRUE will force the calculation of the unassigned eigengene in the returned newMEs, but not in the returned oldMEs. (defualt = True)</p></li>
<li><p><strong>trapErrors</strong> – Controls whether computational errors in calculating module eigengenes, their dissimilarity, and merging trees should be trapped. If TRUE, errors will be trapped and the function will return the input colors. If FALSE, errors will cause the function to stop. (defualt = False)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionaty contains: “colors”: Color labels for the genes corresponding to merged modules. The function attempts to mimic the mode of the input colors: if the input colors is numeric, character and factor, respectively, so is the output. Note, however, that if the fnction performs relabeling, a standard sequence of labels will be used: integers starting at 1 if the input colors is numeric, and a sequence of color labels otherwise. “dendro”: Hierarchical clustering dendrogram (average linkage) of the eigengenes of the most recently computed tree. If iterate was set TRUE, this will be the dendrogram of the merged modules, otherwise it will be the dendrogram of the original modules. “oldDendro”: Hierarchical clustering dendrogram (average linkage) of the eigengenes of the original modules. “cutHeight”: The input cutHeight. “oldMEs”: Module eigengenes of the original modules in the sets given by useSets. “newMEs”: Module eigengenes of the merged modules in the sets given by useSets. “allOK”: A boolean set to TRUE.</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>trapErrors==TRUE</strong> – A dictionaty contains: “colors”: A copy of the input colors. “allOK”: a boolean set to FALSE.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.moduleEigengenes">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">moduleEigengenes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">expr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">impute</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nPC</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'along</span> <span class="pre">average'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">excludeGrey</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grey</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'grey'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subHubs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">softPower</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scaleVar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trapErrors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.moduleEigengenes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.moduleEigengenes" title="Link to this definition">¶</a></dt>
<dd><p>Calculates module eigengenes (1st principal component) of modules in a given single dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>expr</strong> (<em>pandas dataframe</em>) – Expression data for a single set in the form of a data frame where rows are samples and columns are genes (probes).</p></li>
<li><p><strong>colors</strong> (<em>list</em>) – A list of the same length as the number of probes in expr, giving module color for all probes (genes). Color “grey” is reserved for unassigned genes.</p></li>
<li><p><strong>impute</strong> (<em>bool</em>) – If TRUE, expression data will be checked for the presence of NA entries and if the latter are present, numerical data will be imputed. (defualt = True)</p></li>
<li><p><strong>nPC</strong> (<em>int</em>) – Number of principal components and variance explained entries to be calculated. Note that only the first principal component is returned; the rest are used only for the calculation of proportion of variance explained. If given nPC is greater than 10, a warning is issued. (default = 1)</p></li>
<li><p><strong>align</strong> (<em>str</em>) – Controls whether eigengenes, whose orientation is undetermined, should be aligned with average expression (align = “along average”) or left as they are (align = “”). Any other value will trigger an error. (default = “along average”)</p></li>
<li><p><strong>excludeGrey</strong> (<em>bool</em>) – Should the improper module consisting of ‘grey’ genes be excluded from the eigengenes (default = False)</p></li>
<li><p><strong>grey</strong> (<em>str</em>) – Value of colors designating the improper module. Note that if colors is a factor of numbers, the default value will be incorrect. (default = grey)</p></li>
<li><p><strong>subHubs</strong> (<em>bool</em>) – Controls whether hub genes should be substituted for missing eigengenes. If TRUE, each missing eigengene (i.e., eigengene whose calculation failed and the error was trapped) will be replaced by a weighted average of the most connected hub genes in the corresponding module. If this calculation fails, or if subHubs==FALSE, the value of trapErrors will determine whether the offending module will be removed or whether the function will issue an error and stop. (default = True)</p></li>
<li><p><strong>softPower</strong> (<em>int</em>) – The power used in soft-thresholding the adjacency matrix. Only used when the hubgene approximation is necessary because the principal component calculation failed. It must be non-negative. The default value should only be changed if there is a clear indication that it leads to incorrect results. (default = 6)</p></li>
<li><p><strong>trapErrors</strong> (<em>bool</em>) – Controls handling of errors from that may arise when there are too many NA entries in expression data. If TRUE, errors from calling these functions will be trapped without abnormal exit. If FALSE, errors will cause the function to stop. Note, however, that subHubs takes precedence in the sense that if subHubs==TRUE and trapErrors==FALSE, an error will be issued only if both the principal component and the hubgene calculations have failed. (default = False)</p></li>
<li><p><strong>scaleVar</strong> (<em>bool</em>) – can be used to turn off scaling of the expression data before calculating the singular value decomposition. The scaling should only be turned off if the data has been scaled previously, in which case the function can run a bit faster. Note however that the function first imputes, then scales the expression data in each module. If the expression contain missing data, scaling outside of the function and letting the function impute missing data may lead to slightly different results than if the data is scaled within the function. (default = True)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary containing: “eigengenes”: Module eigengenes in a dataframe, with each column corresponding to one eigengene. The columns are named by the corresponding color with an “ME” prepended, e.g., MEturquoise etc. If returnValidOnly==FALSE, module eigengenes whose calculation failed have all components set to NA. “averageExpr”: If align == “along average”, a dataframe containing average normalized expression in each module. The columns are named by the corresponding color with an “AE” prepended, e.g., AEturquoise etc. “varExplained”: A dataframe in which each column corresponds to a module, with the component varExplained[PC, module] giving the variance of module module explained by the principal component no. PC. The calculation is exact irrespective of the number of computed principal components. At most 10 variance explained values are recorded in this dataframe. “nPC”: A copy of the input nPC. “validMEs”: A boolean vector. Each component (corresponding to the columns in data) is TRUE if the corresponding eigengene is valid, and FALSE if it is invalid. Valid eigengenes include both principal components and their hubgene approximations. When returnValidOnly==FALSE, by definition all returned eigengenes are valid and the entries of validMEs are all TRUE. “validColors”: A copy of the input colors with entries corresponding to invalid modules set to grey if given, otherwise 0 if colors is numeric and “grey” otherwise. “allOK”: Boolean flag signalling whether all eigengenes have been calculated correctly, either as principal components or as the hubgene average approximation. “allPC”: Boolean flag signalling whether all returned eigengenes are principal components. “isPC”: Boolean vector. Each component (corresponding to the columns in eigengenes) is TRUE if the corresponding eigengene is the first principal component and FALSE if it is the hubgene approximation or is invalid. “isHub”: Boolean vector. Each component (corresponding to the columns in eigengenes) is TRUE if the corresponding eigengene is the hubgene approximation and FALSE if it is the first principal component or is invalid. “validAEs”: Boolean vector. Each component (corresponding to the columns in eigengenes) is TRUE if the corresponding module average expression is valid. “allAEOK”: Boolean flag signalling whether all returned module average expressions contain valid data. Note that returnValidOnly==TRUE does not imply allAEOK==TRUE: some invalid average expressions may be returned if their corresponding eigengenes have been calculated correctly.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.module_trait_relationships_heatmap">
<span class="sig-name descname"><span class="pre">module_trait_relationships_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">metaData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'module-traitRelationships'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.module_trait_relationships_heatmap"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.module_trait_relationships_heatmap" title="Link to this definition">¶</a></dt>
<dd><p>plot topic-trait relationship heatmap</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>metaData</strong> (<em>list</em>) – traits you would like to see the relationship with topics (must be column name of datExpr.obs)</p></li>
<li><p><strong>show</strong> (<em>bool</em>) – indicate if you want to show the plot or not (default: True)</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name and path of the plot use for save (default: topic-traitRelationships)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.multiSetMEs">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">multiSetMEs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exprData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colors</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">universalColors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useGenes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">impute</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nPC</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">align</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'along</span> <span class="pre">average'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">excludeGrey</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subHubs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trapErrors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">softPower</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grey</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.multiSetMEs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.multiSetMEs" title="Link to this definition">¶</a></dt>
<dd><p>Calculates module eigengenes for several sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exprData</strong> (<em>pandas dataframe</em>) – Expression data in a multi-set format</p></li>
<li><p><strong>colors</strong> (<em>list</em>) – A list of the same length as the number of probes in expr, giving module color for all probes (genes). Color “grey” is reserved for unassigned genes.</p></li>
<li><p><strong>universalColors</strong> (<em>list</em>) – Alternative specification of module assignment</p></li>
<li><p><strong>useSets</strong> (<em>list</em>) – If calculations are requested in (a) selected set(s) only, the set(s) can be specified here. Defaults to all sets.</p></li>
<li><p><strong>useGenes</strong> (<em>list</em>) – Can be used to restrict calculation to a subset of genes</p></li>
<li><p><strong>impute</strong> (<em>bool</em>) – If TRUE, expression data will be checked for the presence of NA entries and if the latter are present, numerical data will be imputed. (defualt = True)</p></li>
<li><p><strong>nPC</strong> (<em>int</em>) – Number of principal components and variance explained entries to be calculated. Note that only the first principal component is returned; the rest are used only for the calculation of proportion of variance explained. If given nPC is greater than 10, a warning is issued. (default = 1)</p></li>
<li><p><strong>align</strong> (<em>str</em>) – Controls whether eigengenes, whose orientation is undetermined, should be aligned with average expression (align = “along average”) or left as they are (align = “”). Any other value will trigger an error. (default = “along average”)</p></li>
<li><p><strong>excludeGrey</strong> (<em>bool</em>) – Should the improper module consisting of ‘grey’ genes be excluded from the eigengenes (default = False)</p></li>
<li><p><strong>subHubs</strong> (<em>bool</em>) – Controls whether hub genes should be substituted for missing eigengenes. If TRUE, each missing eigengene (i.e., eigengene whose calculation failed and the error was trapped) will be replaced by a weighted average of the most connected hub genes in the corresponding module. If this calculation fails, or if subHubs==FALSE, the value of trapErrors will determine whether the offending module will be removed or whether the function will issue an error and stop. (default = True)</p></li>
<li><p><strong>trapErrors</strong> (<em>bool</em>) – Controls handling of errors from that may arise when there are too many NA entries in expression data. If TRUE, errors from calling these functions will be trapped without abnormal exit. If FALSE, errors will cause the function to stop. Note, however, that subHubs takes precedence in the sense that if subHubs==TRUE and trapErrors==FALSE, an error will be issued only if both the principal component and the hubgene calculations have failed. (default = False)</p></li>
<li><p><strong>softPower</strong> (<em>int</em>) – The power used in soft-thresholding the adjacency matrix. Only used when the hubgene approximation is necessary because the principal component calculation failed. It must be non-negative. The default value should only be changed if there is a clear indication that it leads to incorrect results. (default = 6)</p></li>
<li><p><strong>grey</strong> (<em>str</em>) – Value of colors or universalColors (whichever applies) designating the improper module</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary similar in spirit to the input exprData</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.orderMEs">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">orderMEs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">MEs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">greyLast</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">greyName</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'MEgrey'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orderBy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">order</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">useSets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.orderMEs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.orderMEs" title="Link to this definition">¶</a></dt>
<dd><p>Reorder given (eigen-)vectors such that similar ones (as measured by correlation) are next to each other.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>MEs</strong> (<em>dict</em>) – Module eigengenes in a multi-set format.</p></li>
<li><p><strong>greyLast</strong> – Normally the color grey is reserved for unassigned genes; hence the grey module is not a proper module and it is conventional to put it last. If this is not desired, set the parameter to FALSE. (default = True)</p></li>
</ul>
</dd>
</dl>
<p>:type greyLast:bool
:param greyName: Name of the grey module eigengene. (default = “MEgrey”)
:type greyName: str
:param orderBy: Specifies the set by which the eigengenes are to be ordered (in all other sets as well). Defaults to the first set in useSets (or the first set, if useSets is not given). (defualt = 0)
:type orderBy: int
:param order: Allows the user to specify a custom ordering.
:type order: list
:param useSets: Allows the user to specify for which sets the eigengene ordering is to be performed.
:type useSets: list</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A dictionary of the same type as MEs containing the re-ordered eigengenes.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.pickSoftThreshold">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pickSoftThreshold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dataIsExpr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">RsquaredCut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.9</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">MeanCut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">powerVector</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nBreaks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blockSize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">corOptions</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">networkType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'unsigned'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">moreNetworkConcepts</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gcInterval</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.pickSoftThreshold"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.pickSoftThreshold" title="Link to this definition">¶</a></dt>
<dd><p>Analysis of scale free topology for multiple soft thresholding powers.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> – expression data in a matrix or data frame. Rows correspond to samples and columns to genes.</p></li>
<li><p><strong>data</strong> – pandas dataframe</p></li>
<li><p><strong>dataIsExpr</strong> (<em>bool</em>) – should the data be interpreted as expression (or other numeric) data, or as a similarity matrix of network nodes?</p></li>
<li><p><strong>weights</strong> (<em>pandas dataframe</em>) – optional observation weights for data to be used in correlation calculation. A matrix of the same dimensions as datExpr, containing non-negative weights. Only used with Pearson correlation.</p></li>
<li><p><strong>RsquaredCut</strong> (<em>float</em>) – desired minimum scale free topology fitting index (R^2). (default = 0.9)</p></li>
<li><p><strong>MeanCut</strong> (<em>int</em>) – desired maximum mean connectivity scale free topology fitting index. (default = 100)</p></li>
<li><p><strong>powerVector</strong> (<em>list</em><em> of </em><em>int</em>) – A list of soft thresholding powers for which the scale free topology fit indices are to be calculated.</p></li>
<li><p><strong>nBreaks</strong> (<em>int</em>) – number of bins in connectivity histograms (default = 10)</p></li>
<li><p><strong>blockSize</strong> (<em>int</em>) – block size into which the calculation of connectivity should be broken up. If not given, a suitable value will be calculated using function blockSize and printed if verbose&gt;0. If R runs into memory problems, decrease this value.</p></li>
<li><p><strong>corOptions</strong> (<em>list</em>) – a list giving further options to the correlation function specified in corFnc.</p></li>
<li><p><strong>networkType</strong> (<em>str</em>) – network type. Allowed values are (unique abbreviations of) “unsigned”, “signed”, “signed hybrid”. (default = unsigned)</p></li>
<li><p><strong>moreNetworkConcepts</strong> (<em>bool</em>) – should additional network concepts be calculated? If TRUE, the function will calculate how the network density, the network heterogeneity, and the network centralization depend on the power. For the definition of these additional network concepts, see Horvath and Dong (2008). PloS Comp Biol.</p></li>
<li><p><strong>gcInterval</strong> (<em>int</em>) – a number specifying in interval (in terms of individual genes) in which garbage collection will be performed. The actual interval will never be less than blockSize.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>tuple including powerEstimate: estimate of an appropriate soft-thresholding power which is the lowest power for which the scale free topology fit (R^2) exceeds RsquaredCut and conectivity is less than MeanCut. If (R^2) is below RsquaredCut for all powers maximum will re returned and datout which is a data frame containing the fit indices for scale free topology. The columns contain the soft-thresholding power, adjusted (R^2) for the linear fit, the linear coefficient, adjusted (R^2) for a more complicated fit models, mean connectivity, median connectivity and maximum connectivity. If input moreNetworkConcepts is TRUE, 3 additional columns containing network density, centralization, and heterogeneity.</p>
</dd>
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int and pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.plotModuleEigenGene">
<span class="sig-name descname"><span class="pre">plotModuleEigenGene</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">moduleName</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metadata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.plotModuleEigenGene"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.plotModuleEigenGene" title="Link to this definition">¶</a></dt>
<dd><p>plot module eigen gene figure in given module</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>moduleName</strong> (<em>str</em>) – module name</p></li>
<li><p><strong>metadata</strong> (<em>list</em>) – list of metadata you want to be plotted</p></li>
<li><p><strong>show</strong> (<em>bool</em>) – indicate if you want to see plots in when you run your code</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.preprocess">
<span class="sig-name descname"><span class="pre">preprocess</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.preprocess"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.preprocess" title="Link to this definition">¶</a></dt>
<dd><p>Preprocessing PyWGCNA object including removing obvious outlier on genes and samples</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.replaceMissing">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">replaceMissing</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">replaceWith</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.replaceMissing"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.replaceMissing" title="Link to this definition">¶</a></dt>
<dd><p>Replacing missing (NA) value with appropriate value (for integer number replace with 0 and for string replace with “”)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>x</strong> (<em>object</em>) – value want to replace (single item)</p></li>
<li><p><strong>replaceWith</strong> (<em>object</em>) – define character you want to replace na value by looking at type of data</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>object without any missing (NA) value</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.request_PPI">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">request_PPI</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.request_PPI"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.request_PPI" title="Link to this definition">¶</a></dt>
<dd><p>Getting all the STRING interaction partners of the protein set</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>genes</strong> (<em>list</em>) – list of genes you want to find interaction for</p></li>
<li><p><strong>species</strong> (<em>int</em>) – NCBI taxon identifiers (e.g. Human is 9606, see: <a class="reference external" href="https://string-db.org/cgi/input.pl?input_page_active_form=organisms">https://string-db.org/cgi/input.pl?input_page_active_form=organisms</a>).</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>dataframe contains genes interact with each other</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.request_PPI_image">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">request_PPI_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">request_url</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'https://version-11-5.string-db.org/api/image/network'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.request_PPI_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.request_PPI_image" title="Link to this definition">¶</a></dt>
<dd><p>plot PPI interaction along with link that direct you to the STRING webpage</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>params</strong> (<em>dict</em>) – parameters for requesting</p></li>
<li><p><strong>genes</strong> (<em>list</em>) – list of genes you want to find interaction for</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name of the output file</p></li>
<li><p><strong>request_url</strong> (<em>str</em>) – suitable url for using STRING API</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.request_PPI_subset">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">request_PPI_subset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">request_url</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'https://version-11-5.string-db.org/api/tsv-no-header/interaction_partners'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.request_PPI_subset"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.request_PPI_subset" title="Link to this definition">¶</a></dt>
<dd><p>request STRING to find genes interact with our gene list base</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>request_url</strong> (<em>str</em>) – suitable url for using STRING API</p></li>
<li><p><strong>params</strong> (<em>dict</em>) – parameters for requesting</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>dataframe contains genes interact with each other</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.runWGCNA">
<span class="sig-name descname"><span class="pre">runWGCNA</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.runWGCNA"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.runWGCNA" title="Link to this definition">¶</a></dt>
<dd><p>Preprocess and find modules</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.saveWGCNA">
<span class="sig-name descname"><span class="pre">saveWGCNA</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.saveWGCNA"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.saveWGCNA" title="Link to this definition">¶</a></dt>
<dd><p>Saves the current WGCNA in pickle format with the .p extension</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.scaleFreeFitIndex">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">scaleFreeFitIndex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">k</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nBreaks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.scaleFreeFitIndex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.scaleFreeFitIndex" title="Link to this definition">¶</a></dt>
<dd><p>calculates several indices (fitting statistics) for evaluating scale free topology fit.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>k</strong> (<em>list</em>) – numeric list whose components contain non-negative values</p></li>
<li><p><strong>nBreaks</strong> (<em>int</em>) – (default = 10)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.setMetadataColor">
<span class="sig-name descname"><span class="pre">setMetadataColor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">col</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmap</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.setMetadataColor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.setMetadataColor" title="Link to this definition">¶</a></dt>
<dd><p>set color pallete for each group of metadata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>col</strong> (<em>str</em>) – name of metadata</p></li>
<li><p><strong>cmap</strong> (<em>list</em>) – color pallet</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.softConnectivity">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">softConnectivity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datExpr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">corOptions=Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type='unsigned'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">power=6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blockSize=1500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minNSamples=None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.softConnectivity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.softConnectivity" title="Link to this definition">¶</a></dt>
<dd><p>Given expression data or a similarity, the function constructs the adjacency matrix and for each node calculates its connectivity, that is the sum of the adjacency to the other nodes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>datExpr</strong> (<em>pandas dataframe</em>) – a data frame containing the expression data, with rows corresponding to samples and columns to genes.</p></li>
<li><p><strong>corOptions</strong> (<em>pandas dataframe</em>) – character string giving further options to be passed to the correlation function.</p></li>
<li><p><strong>weights</strong> (<em>pandas dataframe</em>) – optional observation weights for datExpr to be used in correlation calculation. A matrix of the same dimensions as datExpr, containing non-negative weights. Only used with Pearson correlation.</p></li>
<li><p><strong>type</strong> (<em>str</em>) – network type. Allowed values are (unique abbreviations of) “unsigned”, “signed”, “signed hybrid”.</p></li>
<li><p><strong>power</strong> (<em>int</em>) – soft thresholding power.</p></li>
<li><p><strong>blockSize</strong> (<em>int</em>) – block size in which adjacency is to be calculated. Too low (say below 100) may make the calculation inefficient, while too high may cause R to run out of physical memory and slow down the computer. Should be chosen such that an array of doubles of size (number of genes) * (block size) fits into available physical memory.</p></li>
<li><p><strong>minNSamples</strong> (<em>int</em>) – minimum number of samples available for the calculation of adjacency for the adjacency to be considered valid. If not given, defaults to the greater of ..minNSamples (currently 4) and number of samples divided by 3. If the number of samples falls below this threshold, the connectivity of the corresponding gene will be returned as NA.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A list with one entry per gene giving the connectivity of each gene in the weighted network.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.top_n_hub_genes">
<span class="sig-name descname"><span class="pre">top_n_hub_genes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">moduleName</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.top_n_hub_genes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.top_n_hub_genes" title="Link to this definition">¶</a></dt>
<dd><p>find top n hub genes based on connectivity in given module</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>moduleName</strong> (<em>str</em>) – name of module you want to top n hub genes</p></li>
<li><p><strong>n</strong> (<em>int</em>) – number of top hub genes</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>dataframe contains top n hun genes along with connectivity score and additional gene information you added to your expression matrix</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.updateGeneInfo">
<span class="sig-name descname"><span class="pre">updateGeneInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">geneInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.updateGeneInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.updateGeneInfo" title="Link to this definition">¶</a></dt>
<dd><p>add/update genes info in datExpr and geneExpr anndata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>geneInfo</strong> (<em>pandas dataframe</em>) – gene information table you want to add to your data</p></li>
<li><p><strong>path</strong> (<em>str</em>) – path of geneInfo</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – separation symbol to use for reading data in path properly (default: “,”)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.wgcna.WGCNA.updateSampleInfo">
<span class="sig-name descname"><span class="pre">updateSampleInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sampleInfo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sep</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">','</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/wgcna.html#WGCNA.updateSampleInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.wgcna.WGCNA.updateSampleInfo" title="Link to this definition">¶</a></dt>
<dd><p>add/update metadata in datExpr and geneExpr anndata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sampleInfo</strong> (<em>pandas dataframe</em>) – Sample information table you want to add to your data</p></li>
<li><p><strong>path</strong> (<em>str</em>) – path of metaData</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – separation symbol to use for reading data in path properly (default: “,”)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class" id="module-PyWGCNA.comparison">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">PyWGCNA.comparison.</span></span><span class="sig-name descname"><span class="pre">Comparison</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">geneModules</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison" title="Link to this definition">¶</a></dt>
<dd><p>A class used to compare PyWGCNA to another PyWGCNA or any gene marker table</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>geneModules</strong> (<em>dict</em>) – gene modules of networks</p></li>
<li><p><strong>jaccard_similarity</strong> (<em>pandas dataframe</em>) – jaccard similarity of common genes between each modules</p></li>
<li><p><strong>P_value</strong> (<em>pandas dataframe</em>) – P value of common genes between each modules</p></li>
<li><p><strong>fraction</strong> (<em>pandas dataframe</em>) – fraction of common genes between each modules</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.calculateFraction">
<span class="sig-name descname"><span class="pre">calculateFraction</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.calculateFraction"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.calculateFraction" title="Link to this definition">¶</a></dt>
<dd><p>Calculate common fraction along multiple networks</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>dataframe containing fraction between all modules in all netwroks</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.calculateJaccardSimilarity">
<span class="sig-name descname"><span class="pre">calculateJaccardSimilarity</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.calculateJaccardSimilarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.calculateJaccardSimilarity" title="Link to this definition">¶</a></dt>
<dd><p>Calculate jaccard similarity matrix along multiple networks</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>dataframe containing jaccard similarity between all modules in all PyWGCNA objects</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.calculatePvalue">
<span class="sig-name descname"><span class="pre">calculatePvalue</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.calculatePvalue"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.calculatePvalue" title="Link to this definition">¶</a></dt>
<dd><p>Calculate pvalue of fraction along multiple networks</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>dataframe containing pvalue between all modules in all netwroks</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.compareNetworks">
<span class="sig-name descname"><span class="pre">compareNetworks</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.compareNetworks"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.compareNetworks" title="Link to this definition">¶</a></dt>
<dd><p>compare Networks</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.jaccard">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">jaccard</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">list1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">list2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.jaccard"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.jaccard" title="Link to this definition">¶</a></dt>
<dd><p>Calculate jaccard similarity matrix for two lists</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>list1</strong> (<em>list</em>) – first list containing the data</p></li>
<li><p><strong>list2</strong> (<em>list</em>) – second list containing the data</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>jaccard similarity</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>double</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.plotBubbleComparison">
<span class="sig-name descname"><span class="pre">plotBubbleComparison</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bubble_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'jaccard_similarity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">order1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">order2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'png'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'bubble_comparison'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.plotBubbleComparison"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.plotBubbleComparison" title="Link to this definition">¶</a></dt>
<dd><p>plot comparison matrix as a bubble plot</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bubble_size</strong> (<em>str</em>) – which information you want to use for size of bubble (options: jaccard_similarity or fraction) default: jaccard_similarity</p></li>
<li><p><strong>cutoff</strong> (<em>double</em>) – threshold you used for defining significant comparison</p></li>
<li><p><strong>color</strong> (<em>dict</em>) – if you want to color tick labels for each networks separately</p></li>
<li><p><strong>order1</strong> (<em>list</em><em> of </em><em>str</em>) – order of modules in PyWGCNA1 you want to show in plot (name of each elements should mapped the name of modules in your first PyWGCNA)</p></li>
<li><p><strong>order2</strong> (<em>list</em><em> of </em><em>str</em>) – order of modules in PyWGCNA2 you want to show in plot (name of each elements should mapped the name of modules in your second PyWGCNA)</p></li>
<li><p><strong>figsize</strong> (<em>tuple</em><em> of </em><em>int</em>) – indicate the size of plot (default is base on the number of modules)</p></li>
<li><p><strong>save</strong> (<em>bool</em>) – if you want to save plot as comparison.png near to your script</p></li>
<li><p><strong>save</strong> – indicate if you want to save the plot or not (default: True)</p></li>
<li><p><strong>plot_show</strong> (<em>bool</em>) – indicate if you want to show the plot or not (default: True)</p></li>
<li><p><strong>plot_format</strong> (<em>str</em>) – indicate the format of plot (default: png)</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name and path of the plot use for save (default: jaccard_similarity)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.plotHeatmapComparison">
<span class="sig-name descname"><span class="pre">plotHeatmapComparison</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'jaccard_similarity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_cluster</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_cluster</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'heatmap_comparison'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.plotHeatmapComparison"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.plotHeatmapComparison" title="Link to this definition">¶</a></dt>
<dd><p>plot heatmap comparison</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>color</strong> (<em>str</em>) – how to color heatmap (options: jaccard_similarity or fraction) default: jaccard_similarity</p></li>
<li><p><strong>row_cluster</strong> (<em>bool</em>) – If True, cluster the rows. (default True)</p></li>
<li><p><strong>col_cluster</strong> (<em>bool</em>) – If True, cluster the columns. (default True)</p></li>
<li><p><strong>save</strong> (<em>bool</em>) – if you want to save plot as comparison.png near to your script</p></li>
<li><p><strong>plot_show</strong> (<em>bool</em>) – indicate if you want to show the plot or not (default: True)</p></li>
<li><p><strong>plot_format</strong> (<em>str</em>) – indicate the format of plot (default: pdf)</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name and path of the plot use for save (default: heatmap_comparison)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.plotJaccardSimilarity">
<span class="sig-name descname"><span class="pre">plotJaccardSimilarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'png'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'jaccard_similarity'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.plotJaccardSimilarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.plotJaccardSimilarity" title="Link to this definition">¶</a></dt>
<dd><p>Plot jaccard similarity matrix as a network</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>color</strong> (<em>dict</em>) – if you want to color nodes for each networks separately</p></li>
<li><p><strong>cutoff</strong> (<em>double</em>) – threshold you used for filtering jaccard similarity</p></li>
<li><p><strong>figsize</strong> (<em>tuple</em><em> of </em><em>int</em>) – indicate the size of plot (default is base on the number of nodes that pass cutoff)</p></li>
<li><p><strong>save</strong> (<em>bool</em>) – indicate if you want to save the plot or not (default: True)</p></li>
<li><p><strong>plot_show</strong> (<em>bool</em>) – indicate if you want to show the plot or not (default: True)</p></li>
<li><p><strong>plot_format</strong> (<em>str</em>) – indicate the format of plot (default: png)</p></li>
<li><p><strong>file_name</strong> (<em>str</em>) – name and path of the plot use for save (default: jaccard_similarity)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="PyWGCNA.comparison.Comparison.saveComparison">
<span class="sig-name descname"><span class="pre">saveComparison</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'comparison'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/comparison.html#Comparison.saveComparison"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.comparison.Comparison.saveComparison" title="Link to this definition">¶</a></dt>
<dd><p>save comparison object as comparison.p near to the script</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>name</strong> (<em>str</em>) – name of the pickle file (default: comparison.p)</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function" id="module-PyWGCNA.utils">
<dt class="sig sig-object py" id="PyWGCNA.utils.compareNetworks">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">compareNetworks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">PyWGCNAs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#compareNetworks"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.compareNetworks" title="Link to this definition">¶</a></dt>
<dd><p>Compare serveral PyWGCNA objects</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>PyWGCNAs</strong> (<em>list</em><em> of </em><em>PyWGCNA class</em>) – list of PyWGCNA objects</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>compare object</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Compare class</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="PyWGCNA.utils.compareSingleCell">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">compareSingleCell</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">PyWGCNAs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sc</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#compareSingleCell"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.compareSingleCell" title="Link to this definition">¶</a></dt>
<dd><p>Compare WGCNA and gene marker from single cell experiment</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>PyWGCNAs</strong> (<em>PyWGCNA class</em>) – WGCNA object</p></li>
<li><p><strong>sc</strong> (<em>pandas dataframe</em>) – gene marker table which has ….</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>compare object</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Compare class</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="PyWGCNA.utils.getGeneList">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">getGeneList</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'mmusculus_gene_ensembl'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">attributes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['ensembl_gene_id',</span> <span class="pre">'external_gene_name',</span> <span class="pre">'gene_biotype']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['gene_id',</span> <span class="pre">'gene_name',</span> <span class="pre">'go_id']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">server_domain</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'http://ensembl.org/biomart'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#getGeneList"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.getGeneList" title="Link to this definition">¶</a></dt>
<dd><p>get table that map gene ensembl id to gene name from biomart</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<em>string</em>) – name of the dataset we used from biomart; mouse: mmusculus_gene_ensembl and human: hsapiens_gene_ensembl
you can find more information here: <a class="reference external" href="https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/biomaRt/inst/doc/biomaRt.html#selecting-a-biomart-database-and-dataset">https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/biomaRt/inst/doc/biomaRt.html#selecting-a-biomart-database-and-dataset</a></p></li>
<li><p><strong>attributes</strong> (<em>list</em>) – List the types of data we want</p></li>
<li><p><strong>maps</strong> (<em>list</em>) – mapping between attributes and column names of gene information you want to show</p></li>
<li><p><strong>server_domain</strong> (<em>string</em>) – URL of ensembl biomart server that you want to use to pull out the information (options: [‘’, ‘uswest’, ‘asia’])</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>table extracted from biomart related to the datasets including information from attributes</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="PyWGCNA.utils.getGeneListGOid">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">getGeneListGOid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'mmusculus_gene_ensembl'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">attributes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['ensembl_gene_id',</span> <span class="pre">'external_gene_name',</span> <span class="pre">'go_id']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Goid</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'GO:0003700'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">server_domain</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'http://ensembl.org/biomart'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#getGeneListGOid"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.getGeneListGOid" title="Link to this definition">¶</a></dt>
<dd><p>get table that find gene id and gene name to specific Go term from biomart</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<em>string</em>) – name of the dataset we used from biomart; mouse: mmusculus_gene_ensembl and human: hsapiens_gene_ensembl
you can find more information here: <a class="reference external" href="https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/biomaRt/inst/doc/biomaRt.html#selecting-a-biomart-database-and-dataset">https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/biomaRt/inst/doc/biomaRt.html#selecting-a-biomart-database-and-dataset</a></p></li>
<li><p><strong>attributes</strong> (<em>list</em>) – List the types of data we want</p></li>
<li><p><strong>Goid</strong> (<em>list</em><em> or </em><em>str</em>) – GO term id you would like to get genes from them</p></li>
<li><p><strong>server_domain</strong> (<em>string</em>) – URL of ensembl biomart server that you want to use to pull out the inforamtion</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>table extracted from biomart related to the datasets including information from attributes with filtering</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="PyWGCNA.utils.readComparison">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">readComparison</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#readComparison"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.readComparison" title="Link to this definition">¶</a></dt>
<dd><p>Read a comparison from a saved pickle file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>file</strong> (<em>string</em>) – Name / path of comparison object</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>comparison object</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>comparison class</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="PyWGCNA.utils.readWGCNA">
<span class="sig-prename descclassname"><span class="pre">PyWGCNA.utils.</span></span><span class="sig-name descname"><span class="pre">readWGCNA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/PyWGCNA/utils.html#readWGCNA"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#PyWGCNA.utils.readWGCNA" title="Link to this definition">¶</a></dt>
<dd><p>Read a WGCNA from a saved pickle file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>file</strong> (<em>str</em>) – Name / path of WGCNA object</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>PyWGCNA object</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>PyWGCNA class</p>
</dd>
</dl>
</dd></dl>

</section>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2022, Narges Rezaie.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 7.2.6.<br/>
    </p>
  </div>
</footer>
  </body>
</html>