Search.setIndex({"docnames": ["api", "cite", "index", "installation", "suggested_reading", "tutorials"], "filenames": ["api.rst", "cite.rst", "index.rst", "installation.rst", "suggested_reading.rst", "tutorials.rst"], "titles": ["API Documentation", "Citation", "PyWGCNA: A python package for weighted gene co-expression network analysis", "Installation", "Suggested Reading", "Tutorials"], "terms": {"class": 0, "pywgcna": [0, 1, 3, 5], "geneexp": [0, 2], "speci": 0, "none": 0, "level": 0, "gene": [0, 1, 5], "anndata": 0, "geneexppath": 0, "sep": 0, "geneinfo": 0, "sampleinfo": 0, "sourc": 0, "A": [0, 1], "us": [0, 1, 2, 5], "creat": 0, "express": [0, 1], "along": 0, "data": [0, 5], "trait": [0, 2], "includ": [0, 2], "both": 0, "sampl": [0, 2], "inform": 0, "paramet": [0, 5], "str": 0, "you": [0, 2, 4, 5], "i": [0, 2, 3], "e": 0, "mous": 0, "human": 0, "which": 0, "type": 0, "transcript": 0, "default": 0, "format": [0, 5], "should": 0, "pass": 0, "through": 0, "thi": 0, "x": 0, "matrix": 0, "var": 0, "ob": 0, "panda": 0, "datafram": 0, "ar": [0, 4], "row": 0, "column": 0, "path": 0, "separ": 0, "symbol": 0, "read": [0, 2], "properli": 0, "contain": 0, "have": [0, 5], "same": 0, "index": [0, 2], "name": 0, "id": 0, "static": 0, "updategeneinfo": 0, "geneexpr": 0, "add": 0, "updat": 0, "info": 0, "expr": 0, "tabl": 0, "want": 0, "your": [0, 5], "return": 0, "updatesampleinfo": 0, "metadata": 0, "wgcna": [0, 2, 4], "tpmcutoff": 0, "1": 0, "power": 0, "rsquaredcut": 0, "0": 0, "9": [0, 3], "meancut": 0, "100": 0, "networktyp": 0, "sign": 0, "hybrid": 0, "tomtyp": 0, "minmodules": 0, "50": 0, "nacolor": 0, "grei": 0, "cut": 0, "inf": 0, "medissthr": 0, "2": 0, "save": 0, "fals": 0, "outputpath": 0, "figuretyp": 0, "pdf": 0, "do": [0, 2], "weight": [0, 1, 4], "co": [0, 1], "network": [0, 1, 4, 5], "analysi": [0, 1, 4, 5], "we": [0, 4, 5], "visual": [0, 5], "bool": 0, "indic": 0, "result": 0, "import": 0, "step": 0, "figur": 0, "directori": [0, 3], "all": 0, "object": [0, 5], "where": 0, "rau": 0, "script": 0, "datexpr": 0, "preprocess": [0, 5], "int": 0, "off": 0, "remov": 0, "under": 0, "number": 0, "float": 0, "outlier": 0, "By": 0, "don": 0, "t": 0, "ani": 0, "hierarch": 0, "cluster": [0, 2], "list": [0, 5], "differ": [0, 2], "test": 0, "scale": 0, "free": 0, "10": 0, "11": 0, "21": 0, "r": [0, 4], "squaer": 0, "choos": 0, "between": 0, "mean": 0, "connect": 0, "6": 0, "sft": 0, "soft": 0, "threshold": 0, "ha": 0, "each": 0, "can": [0, 2, 5], "unsign": 0, "adjac": 0, "ndarrai": 0, "calcul": [0, 2], "base": 0, "genetre": 0, "averag": 0, "disstom": 0, "topolog": 0, "overlap": 0, "tom": 0, "measur": [0, 2], "linkag": 0, "input": [0, 5], "interconnected": 0, "like": 0, "larg": 0, "modul": [0, 2, 5], "so": 0, "set": 0, "minimum": 0, "size": 0, "rel": 0, "high": 0, "dynamicmod": 0, "similar": 0, "togeth": 0, "color": 0, "identifi": 0, "find": [0, 2, 5], "them": [0, 5], "me": 0, "eigengen": [0, 2], "diss": 0, "datm": 0, "param": 0, "signedkm": 0, "membership": [0, 2], "moduletraitcor": 0, "correl": [0, 2, 4], "moduletraitpvalu": 0, "p": 0, "valu": 0, "extens": 0, "calculatesignedkm": 0, "exprweight": 0, "meweight": 0, "also": [0, 2], "known": 0, "option": 0, "observ": 0, "dimens": 0, "frame": 0, "correspond": 0, "give": 0, "respect": 0, "coexpressionmoduleplot": 0, "numgen": 0, "numconnect": 0, "mintom": 0, "filter": 0, "file_nam": 0, "plot": [0, 5], "coexpress": 0, "given": 0, "show": 0, "keep": 0, "dict": 0, "dictionari": 0, "kei": 0, "determin": 0, "html": 0, "output": 0, "file": 0, "more": [0, 5], "than": [0, 5], "3": [0, 3], "ppi_network": 0, "modulenam": 0, "genelist": 0, "output_format": 0, "imag": 0, "retriev": 0, "an": [0, 4], "string": [0, 5], "neighborhood": 0, "surround": 0, "one": [0, 2], "protein": [0, 5], "ask": 0, "onli": 0, "interact": [0, 5], "ncbi": 0, "taxon": 0, "g": 0, "9606": 0, "see": 0, "http": 0, "db": 0, "org": 0, "cgi": 0, "pl": 0, "input_page_active_form": 0, "organ": 0, "ppi": [0, 5], "highres_imag": 0, "svg": 0, "score": 0, "tomsimilar": 0, "adjmat": 0, "tomdenom": 0, "min": 0, "dissimilar": 0, "from": [0, 2, 5], "squar": 0, "symmetr": 0, "entri": 0, "neg": 0, "allow": 0, "charact": 0, "specifi": 0, "variant": 0, "recogn": 0, "standard": 0, "describ": [0, 4], "zhang": 0, "horvath": 0, "2005": 0, "function": [0, 5], "denomin": 0, "replac": 0, "The": 0, "mai": 0, "produc": 0, "better": 0, "time": 0, "consid": 0, "experiment": 0, "hold": 0, "selectcol": 0, "adjacencytyp": 0, "coropt": 0, "empti": 0, "weightargnam": 0, "distanc": 0, "select": 0, "whose": 0, "either": 0, "numer": 0, "boolean": 0, "uniqu": 0, "abbrevi": 0, "addit": 0, "argument": 0, "corfnc": 0, "non": 0, "pearson": 0, "length": 0, "repres": 0, "variabl": 0, "y": 0, "null": 0, "analysewgcna": 0, "order": 0, "true": 0, "analys": 0, "relationship": 0, "heatmap": 0, "go": [0, 3, 5], "term": 0, "up": 0, "mind": 0, "multipl": 0, "when": [0, 1], "run": [0, 3], "code": 0, "barplotmoduleeigengen": 0, "combin": 0, "colorbar": 0, "bar": 0, "eigen": 0, "praram": 0, "calblocks": 0, "matrixs": 0, "rectangularblock": 0, "maxmemoryalloc": 0, "overheadfactor": 0, "suitabl": 0, "block": 0, "checkadjmat": 0, "max": 0, "check": 0, "correct": 0, "maximum": 0, "rais": 0, "exit": 0, "checkandscaleweight": 0, "scalebymax": 0, "dive": 0, "boll": 0, "process": [0, 5], "checkset": 0, "checkstructur": 0, "useset": 0, "whether": 0, "must": 0, "compon": 0, "content": 0, "arrai": 0, "If": [0, 4], "incorrect": 0, "structur": 0, "trigger": 0, "error": 0, "appropri": 0, "flag": 0, "specif": 0, "some": [0, 5], "nset": 0, "vector": 0, "ngene": 0, "nsampl": 0, "structureok": 0, "equal": 0, "paramt": 0, "few": 0, "its": 0, "otherwis": 0, "exhaust": 0, "meant": 0, "catch": 0, "obviou": 0, "user": [0, 2], "rather": 0, "bulletproof": 0, "checksimilar": 0, "consensusmedissimilaritymajor": 0, "useab": 0, "method": 0, "consensu": 0, "cor": 0, "realiz": 0, "sever": 0, "consensusorderm": 0, "greylast": 0, "greynam": 0, "megrei": 0, "reorder": 0, "ones": 0, "next": 0, "other": 0, "multi": 0, "control": 0, "absolut": 0, "plain": 0, "defualt": 0, "perform": [0, 5], "normal": 0, "reserv": 0, "unassign": 0, "henc": 0, "proper": 0, "convent": 0, "put": 0, "last": 0, "desir": 0, "major": 0, "re": 0, "cutre": 0, "sampletre": 0, "cutheight": 0, "50000": 0, "z": 0, "tree": 0, "scipi": 0, "array_lik": 0, "height": 0, "group": 0, "agglomer": 0, "full": 0, "first": 0, "point": 0, "own": 0, "At": 0, "two": [0, 5], "node": 0, "merg": 0, "final": 0, "singleton": 0, "n_cluster": 0, "cutreehybrid": 0, "dendro": 0, "distm": 0, "minclusters": 0, "20": 0, "deepsplit": 0, "maxcorescatt": 0, "mingap": 0, "maxabscorescatt": 0, "minabsgap": 0, "minsplitheight": 0, "minabssplitheight": 0, "externalbranchsplitfnc": 0, "nexternalsplit": 0, "minexternalsplit": 0, "externalsplitopt": 0, "externalsplitfncneedsdist": 0, "assumesimpleexternalspecif": 0, "pamstag": 0, "pamrespectsdendro": 0, "usemedoid": 0, "maxpamdist": 0, "respectsmallclust": 0, "detect": 0, "dendorgram": 0, "hclust": 0, "wa": 0, "join": 0, "It": 0, "99of": 0, "rang": 0, "5th": 0, "percentil": 0, "dendrogram": 0, "logic": 0, "integ": 0, "4": 0, "provid": 0, "rough": 0, "over": 0, "sensit": 0, "split": 0, "higher": 0, "smaller": 0, "scatter": 0, "core": 0, "branch": 0, "fraction": 0, "gap": 0, "overrid": 0, "below": 0, "automat": 0, "zero": 0, "minabssplith": 0, "abov": 0, "evalu": 0, "singl": [0, 2, 5], "decid": 0, "further": 0, "extern": [0, 2, 5], "need": 0, "element": 0, "per": 0, "scalar": 0, "assum": 0, "simpl": 0, "second": 0, "pam": 0, "stage": 0, "sens": 0, "assign": 0, "lie": 0, "medoid": 0, "closest": 0, "fail": 0, "becaus": 0, "insuffici": 0, "individu": 0, "detail": 0, "detec": 0, "findmodul": 0, "kwarg": 0, "origin": [0, 4], "pipelin": 0, "pick": 0, "5": 0, "dynam": 0, "fixdatastructur": 0, "encapsul": 0, "wrapper": 0, "make": 0, "work": 0, "multiset": 0, "collect": 0, "ot": 0, "oper": 0, "functional_enrichment_analysi": 0, "p_valu": 0, "enrich": [0, 5], "kegg": [0, 5], "reactom": [0, 5], "databas": [0, 5], "tupl": 0, "enrichr": 0, "librari": [0, 2], "": 0, "custom": 0, "defin": 0, "gene_set": 0, "gmt": 0, "here": [0, 1, 2, 5], "maayanlab": 0, "cloud": 0, "stat": 0, "fill": 0, "pvalu": 0, "05": 0, "getdattrait": 0, "get": 0, "getgenemodul": 0, "request": 0, "getmodulenam": 0, "getmodulesgen": 0, "geneid": 0, "goodgenesfun": 0, "usesampl": 0, "usegen": 0, "minfract": 0, "minnsampl": 0, "minngen": 0, "tol": 0, "minrelativeweight": 0, "miss": 0, "varianc": 0, "good": 0, "fit": 0, "actual": 0, "fall": 0, "issu": 0, "small": 0, "compar": [0, 2, 5], "against": 0, "divid": 0, "note": 0, "exclud": 0, "goodsamplesfun": 0, "findmodulesminimum": 0, "goodsamplesgen": 0, "necessari": 0, "iter": 0, "tripl": 0, "goodgen": 0, "goodsampl": 0, "allok": 0, "everyth": 0, "okai": 0, "d": 0, "complet": 0, "analyz": [0, 5], "pdist": 0, "algorithm": 0, "encod": 0, "intramodularconnect": 0, "mat": 0, "intramodular": 0, "within": 0, "label": 0, "ncol": 0, "im": 0, "getwholenetworkconnect": 0, "total": 0, "extra": 0, "modular": 0, "intra": 0, "labels2color": 0, "zeroisgrei": 0, "colorseq": 0, "convert": 0, "sequenc": 0, "mergeclosemodul": 0, "exprdata": 0, "imput": 0, "checkdataformat": 0, "unassdcolor": 0, "equalizequantil": 0, "quantilesummari": 0, "consensusquantil": 0, "relabel": 0, "getnewm": 0, "getnewunassdm": 0, "traperror": 0, "too": 0, "close": 0, "factor": 0, "been": 0, "befor": 0, "comput": 0, "thei": 0, "disabl": 0, "presenc": 0, "na": 0, "caus": 0, "hubgen": 0, "approxim": 0, "usabl": 0, "enter": 0, "quantil": 0, "reproduc": 0, "old": 0, "mani": 0, "least": 0, "achiev": 0, "One": 0, "median": 0, "how": [0, 5], "refer": 0, "qualifi": 0, "procedur": 0, "repeat": 0, "until": 0, "chang": 0, "execut": 0, "after": 0, "packag": [0, 1, 4], "start": [0, 5], "manipul": 0, "doe": 0, "forc": 0, "newm": 0, "oldm": 0, "trap": 0, "stop": 0, "dictionati": 0, "attempt": 0, "mimic": 0, "mode": 0, "howev": 0, "fnction": 0, "most": [0, 2], "recent": [0, 2], "olddendro": 0, "copi": 0, "moduleeigengen": 0, "npc": 0, "align": 0, "excludegrei": 0, "subhub": 0, "softpow": 0, "scalevar": 0, "1st": 0, "princip": 0, "dataset": [0, 2], "form": 0, "probe": 0, "latter": 0, "present": 0, "explain": 0, "rest": 0, "proport": 0, "greater": [0, 3], "warn": 0, "orient": 0, "undetermin": 0, "left": 0, "improp": 0, "consist": 0, "design": [0, 2], "hub": 0, "substitut": 0, "offend": 0, "clear": 0, "lead": 0, "handl": 0, "aris": 0, "call": 0, "without": 0, "abnorm": 0, "take": 0, "preced": 0, "turn": 0, "singular": 0, "decomposit": 0, "previous": 0, "case": 0, "bit": 0, "faster": 0, "outsid": 0, "let": 0, "slightli": 0, "prepend": 0, "meturquois": 0, "etc": 0, "returnvalidonli": 0, "averageexpr": 0, "ae": 0, "aeturquois": 0, "varexplain": 0, "pc": 0, "exact": 0, "irrespect": 0, "record": 0, "validm": 0, "valid": 0, "invalid": 0, "definit": 0, "validcolor": 0, "signal": 0, "correctli": 0, "allpc": 0, "ispc": 0, "ishub": 0, "valida": 0, "allaeok": 0, "impli": 0, "module_trait_relationships_heatmap": 0, "traitrelationship": 0, "topic": 0, "would": 0, "multisetm": 0, "universalcolor": 0, "altern": 0, "restrict": 0, "subset": 0, "whichev": 0, "appli": 0, "spirit": 0, "orderm": 0, "orderbi": 0, "well": 0, "picksoftthreshold": 0, "dataisexpr": 0, "powervector": 0, "nbreak": 0, "blocksiz": 0, "morenetworkconcept": 0, "gcinterv": 0, "topologi": 0, "interpret": 0, "bin": 0, "histogram": 0, "broken": 0, "print": 0, "verbos": 0, "memori": 0, "problem": 0, "decreas": 0, "concept": 0, "densiti": 0, "heterogen": 0, "central": 0, "depend": 0, "For": 0, "dong": 0, "2008": 0, "plo": 0, "comp": 0, "biol": 0, "interv": 0, "garbag": 0, "never": 0, "less": 0, "powerestim": 0, "estim": 0, "lowest": 0, "exce": 0, "conect": 0, "datout": 0, "adjust": 0, "linear": 0, "coeffici": 0, "complic": 0, "model": 0, "plotmoduleeigengen": 0, "replacemiss": 0, "replacewith": 0, "item": 0, "look": 0, "request_ppi": 0, "partner": 0, "request_ppi_imag": 0, "request_url": 0, "version": [0, 3], "link": 0, "direct": 0, "webpag": 0, "url": 0, "request_ppi_subset": 0, "tsv": 0, "header": 0, "interaction_partn": 0, "our": [0, 1], "runwgcna": 0, "savewgcna": 0, "current": 0, "pickl": 0, "scalefreefitindex": 0, "k": 0, "statist": 0, "setmetadatacolor": 0, "col": 0, "cmap": 0, "pallet": 0, "softconnect": 0, "1500": 0, "construct": 0, "sum": 0, "low": 0, "sai": 0, "ineffici": 0, "while": 0, "out": 0, "physic": 0, "slow": 0, "down": 0, "chosen": 0, "doubl": 0, "avail": 0, "top_n_hub_gen": 0, "n": 0, "top": 0, "hun": 0, "ad": 0, "comparison": [0, 2], "genemodul": 0, "anoth": [0, 2], "marker": [0, 2, 5], "jaccard_similar": 0, "jaccard": 0, "common": 0, "calculatefract": 0, "netwrok": 0, "calculatejaccardsimilar": 0, "calculatepvalu": 0, "comparenetwork": [0, 2], "list1": 0, "list2": 0, "plotbubblecomparison": 0, "bubble_s": 0, "cutoff": 0, "01": 0, "order1": 0, "order2": 0, "figsiz": 0, "plot_show": 0, "plot_format": 0, "png": 0, "bubble_comparison": 0, "bubbl": 0, "signific": 0, "tick": 0, "pywgcna1": 0, "map": 0, "pywgcna2": 0, "plotheatmapcomparison": 0, "row_clust": 0, "col_clust": 0, "heatmap_comparison": 0, "plotjaccardsimilar": 0, "savecomparison": 0, "util": 0, "server": 0, "comparesinglecel": [0, 2], "sc": 0, "cell": [0, 2, 5], "experi": 0, "getgenelist": [0, 2], "mmusculus_gene_ensembl": 0, "attribut": 0, "ensembl_gene_id": 0, "external_gene_nam": 0, "gene_biotyp": 0, "gene_id": 0, "gene_nam": 0, "go_id": 0, "server_domain": 0, "ensembl": 0, "biomart": 0, "hsapiens_gene_ensembl": 0, "bioconductor": 0, "riken": 0, "jp": 0, "bioc": 0, "vignett": 0, "inst": 0, "doc": 0, "pull": 0, "uswest": 0, "asia": 0, "extract": 0, "relat": [0, 2], "getgenelistgoid": [0, 2], "goid": 0, "0003700": 0, "inforamt": 0, "readcomparison": [0, 2], "readwgcna": [0, 2], "pleas": 1, "cite": 1, "paper": 1, "rezai": 1, "narg": 1, "fairli": 1, "rees": 1, "ali": 1, "mortazavi": 1, "python": [1, 3], "biorxiv": 1, "2022": 1, "highli": 2, "summar": 2, "methodologi": 2, "github": 2, "page": 2, "instal": 2, "pypi": 2, "recommend": 2, "commit": 2, "tutori": 2, "suggest": 2, "citat": 2, "api": 2, "document": 2, "search": 2, "To": 3, "requir": 3, "releas": 3, "pip": 3, "git": 3, "clone": 3, "repositori": 3, "unfamiliar": 4, "refrenc": 4, "public": 4, "variou": 4, "featur": 4, "program": 4, "languag": 4, "clean": 5, "pre": 5, "quick": 5, "load": 5, "access": 5, "three": 5, "shown": 5, "recov": 5}, "objects": {"PyWGCNA": [[0, 0, 0, "-", "comparison"], [0, 0, 0, "-", "geneExp"], [0, 0, 0, "-", "utils"], [0, 0, 0, "-", "wgcna"]], "PyWGCNA.comparison": [[0, 1, 1, "", "Comparison"]], "PyWGCNA.comparison.Comparison": [[0, 2, 1, "", "calculateFraction"], [0, 2, 1, "", "calculateJaccardSimilarity"], [0, 2, 1, "", "calculatePvalue"], [0, 2, 1, "", "compareNetworks"], [0, 2, 1, "", "jaccard"], [0, 2, 1, "", "plotBubbleComparison"], [0, 2, 1, "", "plotHeatmapComparison"], [0, 2, 1, "", "plotJaccardSimilarity"], [0, 2, 1, "", "saveComparison"]], "PyWGCNA.geneExp": [[0, 1, 1, "", "GeneExp"]], "PyWGCNA.geneExp.GeneExp": [[0, 2, 1, "", "updateGeneInfo"], [0, 2, 1, "", "updateSampleInfo"]], "PyWGCNA.utils": [[0, 3, 1, "", "compareNetworks"], [0, 3, 1, "", "compareSingleCell"], [0, 3, 1, "", "getGeneList"], [0, 3, 1, "", "getGeneListGOid"], [0, 3, 1, "", "readComparison"], [0, 3, 1, "", "readWGCNA"]], "PyWGCNA.wgcna": [[0, 1, 1, "", "WGCNA"]], "PyWGCNA.wgcna.WGCNA": [[0, 2, 1, "", "CalculateSignedKME"], [0, 2, 1, "", "CoexpressionModulePlot"], [0, 2, 1, "", "PPI_network"], [0, 2, 1, "", "TOMsimilarity"], [0, 2, 1, "", "adjacency"], [0, 2, 1, "", "analyseWGCNA"], [0, 2, 1, "", "barplotModuleEigenGene"], [0, 2, 1, "", "calBlockSize"], [0, 2, 1, "", "checkAdjMat"], [0, 2, 1, "", "checkAndScaleWeights"], [0, 2, 1, "", "checkSets"], [0, 2, 1, "", "checkSimilarity"], [0, 2, 1, "", "consensusMEDissimilarityMajor"], [0, 2, 1, "", "consensusOrderMEs"], [0, 2, 1, "", "cutree"], [0, 2, 1, "", "cutreeHybrid"], [0, 2, 1, "", "findModules"], [0, 2, 1, "", "fixDataStructure"], [0, 2, 1, "", "functional_enrichment_analysis"], [0, 2, 1, "", "getDatTraits"], [0, 2, 1, "", "getGeneModule"], [0, 2, 1, "", "getModuleName"], [0, 2, 1, "", "getModulesGene"], [0, 2, 1, "", "goodGenesFun"], [0, 2, 1, "", "goodSamplesFun"], [0, 2, 1, "", "goodSamplesGenes"], [0, 2, 1, "", "hclust"], [0, 2, 1, "", "intramodularConnectivity"], [0, 2, 1, "", "labels2colors"], [0, 2, 1, "", "mergeCloseModules"], [0, 2, 1, "", "moduleEigengenes"], [0, 2, 1, "", "module_trait_relationships_heatmap"], [0, 2, 1, "", "multiSetMEs"], [0, 2, 1, "", "orderMEs"], [0, 2, 1, "", "pickSoftThreshold"], [0, 2, 1, "", "plotModuleEigenGene"], [0, 2, 1, "", "preprocess"], [0, 2, 1, "", "replaceMissing"], [0, 2, 1, "", "request_PPI"], [0, 2, 1, "", "request_PPI_image"], [0, 2, 1, "", "request_PPI_subset"], [0, 2, 1, "", "runWGCNA"], [0, 2, 1, "", "saveWGCNA"], [0, 2, 1, "", "scaleFreeFitIndex"], [0, 2, 1, "", "setMetadataColor"], [0, 2, 1, "", "softConnectivity"], [0, 2, 1, "", "top_n_hub_genes"], [0, 2, 1, "", "updateGeneInfo"], [0, 2, 1, "", "updateSampleInfo"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "function", "Python function"]}, "titleterms": {"api": 0, "document": 0, "citat": 1, "pywgcna": 2, "A": 2, "python": 2, "packag": 2, "weight": 2, "gene": 2, "co": 2, "express": 2, "network": 2, "analysi": 2, "indic": 2, "tabl": 2, "instal": 3, "from": 3, "pypi": 3, "recommend": 3, "most": 3, "recent": 3, "commit": 3, "suggest": 4, "read": 4, "tutori": 5}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.todo": 2, "sphinx.ext.viewcode": 1, "sphinx": 60}, "alltitles": {"API Documentation": [[0, "module-PyWGCNA.geneExp"]], "Citation": [[1, "citation"]], "PyWGCNA: A python package for weighted gene co-expression network analysis": [[2, "pywgcna-a-python-package-for-weighted-gene-co-expression-network-analysis"]], "Indices and tables": [[2, "indices-and-tables"]], "Installation": [[3, "installation"]], "Install from PyPi (recommended)": [[3, "install-from-pypi-recommended"]], "Install with the most recent commits": [[3, "install-with-the-most-recent-commits"]], "Suggested Reading": [[4, "suggested-reading"]], "Tutorials": [[5, "tutorials"]]}, "indexentries": {"calculatesignedkme() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.CalculateSignedKME"]], "coexpressionmoduleplot() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.CoexpressionModulePlot"]], "comparison (class in pywgcna.comparison)": [[0, "PyWGCNA.comparison.Comparison"]], "geneexp (class in pywgcna.geneexp)": [[0, "PyWGCNA.geneExp.GeneExp"]], "ppi_network() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.PPI_network"]], "pywgcna.comparison": [[0, "module-PyWGCNA.comparison"]], "pywgcna.geneexp": [[0, "module-PyWGCNA.geneExp"]], "pywgcna.utils": [[0, "module-PyWGCNA.utils"]], "pywgcna.wgcna": [[0, "module-PyWGCNA.wgcna"]], "tomsimilarity() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.TOMsimilarity"]], "wgcna (class in pywgcna.wgcna)": [[0, "PyWGCNA.wgcna.WGCNA"]], "adjacency() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.adjacency"]], "analysewgcna() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.analyseWGCNA"]], "barplotmoduleeigengene() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.barplotModuleEigenGene"]], "calblocksize() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.calBlockSize"]], "calculatefraction() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.calculateFraction"]], "calculatejaccardsimilarity() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.calculateJaccardSimilarity"]], "calculatepvalue() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.calculatePvalue"]], "checkadjmat() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.checkAdjMat"]], "checkandscaleweights() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.checkAndScaleWeights"]], "checksets() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.checkSets"]], "checksimilarity() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.checkSimilarity"]], "comparenetworks() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.compareNetworks"]], "comparenetworks() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.compareNetworks"]], "comparesinglecell() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.compareSingleCell"]], "consensusmedissimilaritymajor() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.consensusMEDissimilarityMajor"]], "consensusordermes() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.consensusOrderMEs"]], "cutree() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.cutree"]], "cutreehybrid() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.cutreeHybrid"]], "findmodules() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.findModules"]], "fixdatastructure() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.fixDataStructure"]], "functional_enrichment_analysis() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.functional_enrichment_analysis"]], "getdattraits() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.getDatTraits"]], "getgenelist() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.getGeneList"]], "getgenelistgoid() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.getGeneListGOid"]], "getgenemodule() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.getGeneModule"]], "getmodulename() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.getModuleName"]], "getmodulesgene() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.getModulesGene"]], "goodgenesfun() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.goodGenesFun"]], "goodsamplesfun() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.goodSamplesFun"]], "goodsamplesgenes() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.goodSamplesGenes"]], "hclust() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.hclust"]], "intramodularconnectivity() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.intramodularConnectivity"]], "jaccard() (pywgcna.comparison.comparison static method)": [[0, "PyWGCNA.comparison.Comparison.jaccard"]], "labels2colors() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.labels2colors"]], "mergeclosemodules() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.mergeCloseModules"]], "module": [[0, "module-PyWGCNA.comparison"], [0, "module-PyWGCNA.geneExp"], [0, "module-PyWGCNA.utils"], [0, "module-PyWGCNA.wgcna"]], "moduleeigengenes() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.moduleEigengenes"]], "module_trait_relationships_heatmap() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.module_trait_relationships_heatmap"]], "multisetmes() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.multiSetMEs"]], "ordermes() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.orderMEs"]], "picksoftthreshold() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.pickSoftThreshold"]], "plotbubblecomparison() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.plotBubbleComparison"]], "plotheatmapcomparison() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.plotHeatmapComparison"]], "plotjaccardsimilarity() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.plotJaccardSimilarity"]], "plotmoduleeigengene() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.plotModuleEigenGene"]], "preprocess() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.preprocess"]], "readcomparison() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.readComparison"]], "readwgcna() (in module pywgcna.utils)": [[0, "PyWGCNA.utils.readWGCNA"]], "replacemissing() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.replaceMissing"]], "request_ppi() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.request_PPI"]], "request_ppi_image() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.request_PPI_image"]], "request_ppi_subset() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.request_PPI_subset"]], "runwgcna() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.runWGCNA"]], "savecomparison() (pywgcna.comparison.comparison method)": [[0, "PyWGCNA.comparison.Comparison.saveComparison"]], "savewgcna() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.saveWGCNA"]], "scalefreefitindex() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.scaleFreeFitIndex"]], "setmetadatacolor() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.setMetadataColor"]], "softconnectivity() (pywgcna.wgcna.wgcna static method)": [[0, "PyWGCNA.wgcna.WGCNA.softConnectivity"]], "top_n_hub_genes() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.top_n_hub_genes"]], "updategeneinfo() (pywgcna.geneexp.geneexp static method)": [[0, "PyWGCNA.geneExp.GeneExp.updateGeneInfo"]], "updategeneinfo() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.updateGeneInfo"]], "updatesampleinfo() (pywgcna.geneexp.geneexp static method)": [[0, "PyWGCNA.geneExp.GeneExp.updateSampleInfo"]], "updatesampleinfo() (pywgcna.wgcna.wgcna method)": [[0, "PyWGCNA.wgcna.WGCNA.updateSampleInfo"]]}})